<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="A set of easy-to-use functions to analyze end read count data of RiboMethSeq (RNA 2'Ome).">
<title>Package to Analyze RiboMethSeq Data • rRMSAnalyzer</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Package to Analyze RiboMethSeq Data">
<meta property="og:description" content="A set of easy-to-use functions to analyze end read count data of RiboMethSeq (RNA 2'Ome).">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">rRMSAnalyzer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/rRMSAnalyzer.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/custom_anno.html">Create and use custom sites annotation</a>
    <a class="dropdown-item" href="articles/custom_ribo.html">Customizing a RiboClass</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/RibosomeCRCL/rRMSAnalyzer/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="rrmsanalyzer-package-to-analyze-ribomethseq-data">rRMSAnalyzer: package to analyze RiboMethSeq data<a class="anchor" aria-label="anchor" href="#rrmsanalyzer-package-to-analyze-ribomethseq-data"></a>
</h1></div>
<p>RiboMethSeq is an RNAseq-based approach to analyze 2’O-ribose methylation (2’Ome).</p>
<p>rRMSAnalyzer is an R package that provides a set of easy-to-use functions to evaluate 2’Ome levels by computing C-scores from RiboMethSeq read end counts as input.</p>
<p>Available features (version 2) :</p>
<ul>
<li>C-score computation (using either mean or median for the window of neighboring positions)</li>
<li>Batch effect adjustment with CombatSeq</li>
<li>Different visualizations to compare samples or sites</li>
<li>Include a table of annotated human rRNA sites</li>
<li>Export of computed C-scores into a dataframe</li>
<li>Semi-automated quality control report</li>
</ul>
<p>These features are more explained in details here :</p>
<blockquote>
<p><strong>Note</strong> We have also developed a <a href="https://github.com/RibosomeCRCL/ribomethseq-nf" class="external-link">dedicated Nextflow pipeline</a> to process the data from sequencing output (fastq files) to useful raw data for rRMSAnalyzer (read end counts).</p>
</blockquote>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The latest version of rRMSAnalyzer package can be installed from Github with :</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"RibosomeCRCL/rRMSAnalyzer"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RibosomeCRCL/rRMSAnalyzer" class="external-link">rRMSAnalyzer</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/load_ribodata.html">load_ribodata</a></span><span class="op">(</span></span>
<span>              count_path <span class="op">=</span> <span class="st">"/path/to/your/csvfiles/directory/"</span>,</span>
<span>              metadata <span class="op">=</span> <span class="st">"path/to/metadata.csv"</span>,</span>
<span>              metadata_key <span class="op">=</span> <span class="st">"filename"</span>,</span>
<span>              metadata_id <span class="op">=</span> <span class="st">"samplename"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compute the c-score using different parameters,</span></span>
<span><span class="co"># including calculation of the local coverage using the mean instead of the median</span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/compute_cscore.html">compute_cscore</a></span><span class="op">(</span><span class="va">ribo</span>, method <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If necessary, adjust any technical biases using ComBat-Seq.</span></span>
<span><span class="co"># Here, as an example, we use the "library" column in metadata.</span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/adjust_bias.html">adjust_bias</a></span><span class="op">(</span><span class="va">ribo</span>,<span class="st">"library"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot a Principal Component Analysis (PCA) whose colors depend on the "condition" column in metadata</span></span>
<span><span class="fu"><a href="reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo</span>,<span class="st">"condition"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>The “getting started” is available on our website : <a href="https://ribosomecrcl.github.io/rRMSAnalyzer/" class="uri">https://ribosomecrcl.github.io/rRMSAnalyzer/</a></p>
<p>A test dataset (ribo_toy) is included in the package.</p>
</div>
<div class="section level2">
<h2 id="help-bug-reports-and-suggestions">Help, bug reports and suggestions<a class="anchor" aria-label="anchor" href="#help-bug-reports-and-suggestions"></a>
</h2>
<p>To report a bug or any suggestion to improve the package, please let us known by opening a new issue on: <a href="https://github.com/RibosomeCRCL/rRMSAnalyzer/issues" class="external-link uri">https://github.com/RibosomeCRCL/rRMSAnalyzer/issues</a></p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>We would like to thank all our collaborators from Jean-Jacques Diaz Team and the Bioinformatic Platform Gilles Thomas for their advices and suggestions.</p>
</div>
<div class="section level2">
<h2 id="funding">Funding<a class="anchor" aria-label="anchor" href="#funding"></a>
</h2>
<p>This project has been funded by the French Cancer Institute (INCa, PLBIO 2019-138 MARACAS), the SIRIC Program (INCa-DGOS-Inserm_12563 LyRICAN), LabEX program (DEVweCan), the French association Ligue Nationale Contre le Cancer and Synergie Lyon Cancer Foundation.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/RibosomeCRCL/rRMSAnalyzer/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/RibosomeCRCL/rRMSAnalyzer/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing rRMSAnalyzer</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Théo Combe <br><small class="roles"> Author, maintainer </small>  </li>
<li>Hermes Paraqindes <br><small class="roles"> Author </small>  </li>
<li>Janice Kielbassa <br><small class="roles"> Author </small>  </li>
<li>Virginie Marcel <br><small class="roles"> Author </small>  </li>
<li>Émilie Thomas <br><small class="roles"> Author </small>  </li>
<li>Anthony Ferrari <br><small class="roles"> Author </small>  </li>
<li>Fondation Synergie Lyon Cancer <br><small class="roles"> Funder </small>  </li>
<li>INCA <br><small class="roles"> Funder </small>  </li>
<li>SIRIC program <br><small class="roles"> Funder </small>  </li>
<li>LabEX program <br><small class="roles"> Funder </small>  </li>
<li>Ligue Nationale Contre le Cancer <br><small class="roles"> Funder </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Théo Combe, Hermes Paraqindes, Janice Kielbassa, Virginie Marcel, Émilie Thomas, Anthony Ferrari, Fondation Synergie Lyon Cancer, INCA, SIRIC program, LabEX program, Ligue Nationale Contre le Cancer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
