---
title: "Quality Control Report"
output: 
    html_document:
        theme: cosmo
        number_sections: true
        toc: true
        toc_float:
            collapsed: false
toc-title: "rRMSAnalyzer"
    
params:
    library_col: "lib"
    specie: "not specified"
    condition_col: "condition"
    ribo_name: "ribo"
    project_name: "Unnamed project"
    comments: "No comment"
---
```{r include = FALSE}
knitr::opts_chunk$set(echo=FALSE)

all_sites <- length(ribo_to_check[[1]][[1]][[1]])
# Get the number of valid sites
ribom <- extract_data(ribo_to_check)
valid_sites <- sum(complete.cases(ribom))
```

```{=html}
<style type="text/css">
  body{
 font-family: Helvetica;    
}

details{
background-color: #E4F3F3;
}

</style>
```

```{r labrary, message=FALSE, warning=FALSE, echo=FALSE}
library(DT)
library(kableExtra)
```

<details>

<summary>

<b> ℹ️ Summary of input data </b>

</summary>
Project name : **`r params$project_name`**<br>
<br>
Data aligned on: **`r if (params$specie=="human") {"GRCh38 (gencode annotation V46)"} else if (params$specie == "mouse") {"mm39 (gencode version vM35)"} else {"unknown annotation"}`**.<br>
<br>
Analysed by: Allyson Moureaux <br>
Date : **`r Sys.time()`**

RiboClass Name : **`r params$ribo_name`** <br> 
Number of uploaded samples: **`r length(ribo_to_check[[1]])`** samples <br> 
RNA(s) used by the coverage tool: **`r as.character(ribo_to_check[["rna_names"]][["original_name"]])`** <br> 
There are **`r as.character(all_sites)`** genomic positions.<br>
<br>
__**C-score**__ <br>
Method of computation : **`r as.character(ribo_to_check["cscore_method"])`** <br>
Window length : **`r as.character(ribo_to_check["cscore_window"])`** nucleotides <br>
There are **`r as.character(valid_sites)`** genomic positions with a valid C-score in all samples.<br>
<br>
__**ComBat-seq**__<br>
ComBat-seq correction:
**`r ifelse(is.null(ribo_to_check[["combatSeq_count"]]),"No","Yes")`** <br>
Metadata column used for ComBat-seq :
**`r ifelse(is.null(ribo_to_check[["col_used_combatSeq"]]),"None",as.character(ribo_to_check["col_used_combatSeq"]))`**

</details>

# Introduction

## Content

This quality control (QC) reports different metrics to ensure that subsequent analyses are free from bias related to the RiboMethSeq technology.<br><br>

In this report, the 5' end read count data is used to ensure the quality of the sequencing coverage on the samples. Then, this measure is used to ensure technical homogeneity of the cohort by comparing samples with each other. QC verifies that the coverage is uniform and reproducible between samples, eliminating the possibility of biases due to sequencing and outlier samples. Finally, a cohort check using C-score values is performed to verify that subsequent analyses will be free of biases that may affect biological conclusions.

## Project summary

This QC report is generated from the analysis of the project **`r params$project_name`**, which is loaded inside the RiboClass object **`r params$ribo_name`**. This **`r params$ribo_name`** object was generated from **`r length(ribo_to_check[[1]])`** samples and **`r as.character(all_sites)`** genomic positions. This project was aligned on **`r if (params$specie=="human") {"GRCh38 (gencode annotation V46)"} else if (params$specie == "mouse") {"mm39 (gencode version vM35)"} else {"unknown annotation"}`**.

The following (r)RNA were used by the coverage tool: **`r as.character(ribo_to_check[["rna_names"]][["original_name"]])`**.

## Method

**C-score computation** : In the following analyses, the C-Score was computed using the **`r as.character(ribo_to_check["cscore_method"])`** of the **`r as.character(ribo_to_check["cscore_window"])`** neighboring positions (Birkedal et al, 2015; Marchand et al, 2016; Marcel et al, 2021). In the end, there are **`r as.character(valid_sites)`** genomic positions with a valid C-score in all samples. Of note, C-score for genomic positions on RNA’s extremities cannot be computed because of the lack of neighbors in the local window.

**Sequencing biases correction** : The possible technical biases induced by the sequencing step are corrected using the **ComBat-seq** method (Zhang et al, 2020; Paraqindes et al, 2023). In this analysis, the correction is **`r ifelse(is.null(ribo_to_check[["combatSeq_count"]]),"unused","used")`**. Here we used the variable "**`r ifelse(is.null(ribo_to_check[["col_used_combatSeq"]]),"None",as.character(ribo_to_check["col_used_combatSeq"]))`**" to perform this correction.

## Metadata

The following metadata are used in the project **`r params$project_name`** :
```{r, echo=FALSE, warning=FALSE}
qcdata <- ribo_to_check[[2]]
```

```{r, echo=FALSE, warning=FALSE}
datatable(qcdata,
          rownames = FALSE,
          extensions = c("Buttons"),
          options = list(
            pageLength = 20,
            scrollX = TRUE,
            dom = 'Bfrtip',  # Add buttons
            buttons = list(
      list(extend = 'csv', title = "Metadata"),
      list(extend = 'excel', title = "Metadata")
      ) # downloading buttons
    )
)
```
 
# Global coverage

In this section, QC measures that are derived from the end read counts on all sequenced positions (i.e., global coverage) are presented to ensure consistent and reproducible sample coverage. A good level of coverage is essential, because the level of rRNA 2'-O-ribose methylation is estimated from its variations.

##  Coverage distribution

The coverage distribution of absolute end read counts at each genomic position is visualized here using a boxplot for each individual sample, controlling for coverage variation. 

Samples with a median coverage below 100 (ie., log10(100)=2; the blue line), are highlighted in red and should be interpreted with caution or excluded from further analysis.

```{r end read count distribution ,warning=FALSE}
rRMSAnalyzer::boxplot_count(ribo_to_check, params$library_col) + ggplot2::ggtitle("End read counts distribution by sample")
```



## Relative Log Coverage

The Relative Log Coverage (RLC)  is calculated to ensure  the reproducibility of the coverage across all samples.

$$\log_2(\frac{pos_{sample} + 1}{pos_{all}})$$


Where :

-   pos~sample~ : the end read count at a specific position for a specific sample
-   pos~all~ : the end read count median of all samples for the same specific position

The following graph shows the Relative Log Coverage boxplot. The blue lines give the median +/- 2 median absolute deviation (mad) variation. Median outside the blue lines identifies putative outlier samples (highlighted in red). Sample presenting a lower or higher coverage compared to other samples might be considered carefully, or outlier and removed from further biological analyses.

```{r relative log cov}
rRMSAnalyzer::plot_rle(ribo_to_check,show_outlier = TRUE) + ggplot2::ggtitle("Relative Log Coverage")
```

## RNA fraction per sample

The rRNA end read counts distribution is used as an indicator of rRNA quality. Any unusual distribution either across 4 rRNA of one sample or across samples might be considered with caution, as it could impact the C-score robustness.<br>
<br>
The following graph shows stacked barplots representing read count fractions of the total end read counts for each rRNA per sample. Theoretical mean fractions have been computed using 38 samples issued from 6 different human tumoral cell lines and the commercial Human XpressRef Universal Total RNA, which consists in total RNA issued from 20 different human adult and fetal normal major organs (Qiagen). These theoretical mean fractions are given as dotted lines. In the case of rRNAs, most end read counts are assigned to the longest rRNAs, namely the 18S (~ 0.30) and the 28S (~ 0.60).

```{r,warning=FALSE, fig.asp = 0.8, fig.width = 10}
rRMSAnalyzer::plot_counts_fraction(ribo_to_check) + ggplot2::ggtitle("RNA fraction per sample")
```

# Profiles comparison

In this section, the difference between the samples using global coverage and C-scores is explored. By using such global information, technical variabilities are more visible than biological variabilities, allowing us to highlight possible biases due to the experimental process. <br><br>
Of note, rRNA sequences are enriched in GC-stretches and in nucleotide repeats that introducing a bias during read alignment. The coverage profile is therefore not uniform, however, it will be characteristic of the rRNA sequence and reproducible across samples. A sample with an atypical coverage pattern should be treated with caution or excluded from further analysis. <br><br>
It is also possible for samples to cluster in sequencing batches. In this case, ComBat-seq correcting algorithm should be used to avoid biases in subsequent analyses. <br>

## Correlation matrix

Coverage profiles are compared with each other within the cohort. This helps highlight divergent coverage profiles. Pairwise sample correlations of coverage profile are calculated using a Pearson correlation based distance. A heatmap is then generated that summarizes the correlation values for the entire cohort. These values go from 0 (no correlation, indicated in red) to 1 (perfect correlation, indicated in blue).<br><br>
A sample identified as an outlier in the previous analyses is highlighted in red and the origin of the issue is indicated on the top two bars and in the legend (e.g., outlier_coverage_distribution).<br>

```{r get outlier data frame}
qcdata <- rRMSAnalyzer::get_outliers(ribo_to_check)
```

```{r heatmap annotated, fig.asp = 0.9, fig.width = 15}
rRMSAnalyzer::plot_heatmap_annotated(ribo_to_check, qcdata)
```

## Correspondance analysis {.tabset .tabset-pills}

To identify distinct coverage profiles, the coverage profiles of all samples in the cohort are compared. The Correspondance Analysis (COA) plot shows the distance between each sample based on its coverage profile. Samples that are far apart are considered putative outliers. <br>
It is possible to visualise the bias linked to the sequencing batch and to evaluate the performance of the ComBat-seq correction algorithm if it has been applied (Zhang et al, 2020; Paraqindes et al, 2023).<br><br>

### COA-1: by library (batch effect ?)

COA-1 allows to visualize the different libraries and thus putative batch effect that can be resolved using ComBat-seq. <br>

```{r, warning=FALSE}
plot_coa <- rRMSAnalyzer::plot_coa(ribo_to_check,color_col = params$library_col)+ 
  ggplot2::scale_color_manual(values = c("L1" = "#00F", "L2" = "orange", "L3" = "purple", "L4" = "#808000", "L5" = "#000080", "L6" = "pink", "L7" = "#FF00FF", "L8" = "#CCCCFF", "L9" = "#9FE2BF", "L10" = "#DFFF00", "L11" = "#DFAF00", "L12" = "#DFDF00"))
print(plot_coa)
```

### COA-2: by outlier samples

COA-2 allows to visualize the sample depending whether they have been considered as outlier based on 1 or 2 previous graphs (see section 2.1 and 2.2). The outlier level of “0” means that no warning has been raised for this sample.

```{r, warning=FALSE}
#add the column of outlier level in ribo
ribo_to_check$metadata$outlier_level <- qcdata$total_outliers

plot_coa_outliers <- rRMSAnalyzer::plot_coa(ribo_to_check, color_col = "outlier_level")
plot_coa_outliers <- plot_coa_outliers + 
   ggplot2::scale_color_manual(values = c("0" = "darkgrey", "1" = "lightsalmon", "2" = "orange", "3" = "#A91101"))

print(plot_coa_outliers)
```

## {-}

# C-scores analysis

In this section, QCs is based on C-scores calculation, which will be used in the subsequent analysis. This final QC will therefore check that the dataset is free of biases that could affect the biological conclusions.<br>
<br>
Since more than 1.5% of the C-score corresponds to non-biological noise, normalization during C-score calculation should limit the dispersion of the samples based on their C-score at all genomic positions in the whole series. Thus, Principal Component Analysis (PCA) on C-scores helps to identify putative batch effects that can be corrected using the inter-normalization option (ComBat-seq tool) (Zhang et al, 2020; Paraqindes et al, 2023). 
<br> <br>
The graph shows a Principal Component Analysis (PCA) plot illustrating the distance between each sample based on its C-score at all genomic positions. The samples are colored by library. The PCA cannot use genomic positions with an invalid C-score. Thus, it uses **`r as.character(valid_sites)`** genomic positions instead of all **`r as.character(all_sites)`**.
<br>

```{r not corrected cscore PCA, ,warning=FALSE, echo=FALSE}
rRMSAnalyzer::plot_pca(ribo_to_check,params$library_col)+ 
   ggplot2::scale_color_manual(values = c("L1" = "#00FFFF", "L2" = "orange", "L3" = "purple", "L4" = "#808000", "L5" = "#000080", "L6" = "pink", "L7" = "#FF00FF", "L8" = "#CCCCFF", "L9" = "#9FE2BF", "L10" = "#DFFF00", "L11" = "#DFAF00", "L12" = "#DFDF00")) +
    ggplot2::labs(color = "library") + # change legend title
    ggplot2::guides(color = ggplot2::guide_legend(title = "library"), fill = "none", shape = "none") #delete all legend not used
```

# Outlier summary

This table recapitulates the different QC parameters used to identify putative outliers as shown in the graph “3.2.2 COA-2”.

```{r write table, warning=FALSE}
output_dir <- getwd()
write.table(qcdata,file.path(output_dir,"quality_control.csv"),sep="\t",row.names=FALSE,quote = FALSE)
knitr::kable(qcdata, "html") %>% kable_styling("striped") %>% scroll_box(width = "100%", height = "500px")
```

# Conclusion

## Analytical Report

This analysis report was generated for the project **`r params$project_name`**, based on a total of **`r length(ribo_to_check[[1]])`** biological samples.<br>
<br>
Coverage estimations were performed over four reference rRNAs (5S, 5.8S, 18S, and 28S), corresponding to **`r as.character(all_sites)`** genomic positions. Among these, **`r as.character(valid_sites)`** positions exhibited a valid C-score in all samples, calculated using a median-based method with a **`r as.character(ribo_to_check["cscore_window"])`-nucleotide window.**<br>
<br>
**`r ifelse(is.null(ribo_to_check[["combatSeq_count"]]),"No","A")` ComBat-seq** batch correction was applied.<br>
<br>
Outlier detection steps identified **`r sum(qcdata$coverage_quality != "pass")`**  sample(s) based on the global coverage distribution, and **`r sum(qcdata$rle_median_quality == "warning")`** sample(s) as a potential outlier(s) using the Relative Log Coverage (RLC) metric. The sample(s) identified as outlier(s) based on QC metrics should be interpreted with caution in downstream analyses :

```{r print outlier samples}
outlier_summary <- data.frame(outliers = qcdata$samplename[qcdata$total_outliers != 0], row.names = "Sample") # print samples that is outlier at least 1 time
knitr::kable(outlier_summary, "html") %>% kable_styling("striped") %>% scroll_box(width = "60%", height = "500px")
```


## Comments

```{r, echo=FALSE, results='asis'}
cat((params$comments), sep = "\n")
```

# Additional information

## Bibliography

- Birkedal, U., Christensen-Dalsgaard, M., Krogh, N., Sabarinathan, R., Gorodkin, J. and Nielsen, H. (2015), Profiling of Ribose Methylations in RNA by High-Throughput Sequencing. Angew. Chem. Int. Ed., 54: 451-455. https://doi.org/10.1002/anie.201408362

- Marcel, V., Kielbassa, J., Marchand, V., Natchiar, K. S., Paraqindes, H., van Long, F. N., Ayadi, L., Bourguignon-Igel, V., Monaco, P. lo, Monchiet, D., Scott, V., Tonon, L., Bray, S. E., Diot, A., Jordan, L. B., Thompson, A. M., Bourdon, J. C., Dubois, T., Andre, F., … Diaz, J. J. (2021). Ribosomal RNA 2′ O-methylation as a novel layer of inter-tumour heterogeneity in breast cancer. In NAR Cancer (Vol. 3, Issue 1). Oxford University Press. https://doi.org/10.1093/narcan/zcab006 

- Marchand, V., Blanloeil-Oillo, F., Helm, M., & Motorin, Y. (2016). Illumina-based RiboMethSeq approach for mapping of 2'-O-Me residues in RNA. Nucleic acids research, 44(16), e135. https://doi.org/10.1093/nar/gkw547 

- Paraqindes H\*, Mourksi NEH\*, Ballesta S\*, Hedjam J, Bourdelais F, Fenouil T, Picart T, Catez F, Combe T, Ferrari A, Kielbassa J, Thomas E, Tonon L, Viari A, Attignon V, Carrere M, Perrossier J, Giraud S, Vanbelle C, Gabut M, Bergeron D, Scott MS, Castro Vega L, Magne N, Huillard E, Sanson M, Meyronet D, Diaz JJ, Ducray F#, Marcel V\*#, Durand S\*#. IDHwt and IDHmut adult-type diffuse gliomas display distinct alterations in ribosome biogenesis and 2’O-methylation of ribosomal RNA. Neuro Oncol 2023, 25(12) :2191-2206.  
https://doi.org/10.1093/neuonc/noad192

-   Yuqing Zhang, Giovanni Parmigiani, W Evan Johnson, ComBat-seq: batch effect adjustment for RNA-seq count data, NAR Genomics and Bioinformatics, Volume 2, Issue 3, 1 September 2020, lqaa078. [https://doi.org/10.1093/nargab/lqaa078](https://doi.org/10.1093/nargab/lqaa078)


## Citation
**When using this report, please cite : Paraqindes et al, Neuro Oncol 2023**


## License

rRMSAnalyzer<br>
Copyright (C) 2023 Centre de Recherche en Cancérologie de Lyon

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.

## Session info

```{r,echo=TRUE}
sessionInfo()
```


-----------------------
This report has been automatically generated with [rRMSAnalyzer](https://github.com/RibosomeCRCL/rRMSAnalyzer) 2.0.1 (July 2025)