<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>rRMSAnalyzer: package to analyze RiboMethSeq data • rRMSAnalyzer</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="rRMSAnalyzer: package to analyze RiboMethSeq data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">rRMSAnalyzer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/rRMSAnalyzer.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">Articles</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/RibosomeCRCL/rRMSAnalyzer/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="rRMSAnalyzer_files/kePrint-0.0.1/kePrint.js"></script><link href="rRMSAnalyzer_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>rRMSAnalyzer: package to analyze RiboMethSeq data</h1>
                        <h4 data-toc-skip class="author">Théo COMBE,
Hermès PARAQINDES, Allyson MOUREAUX, Janice KIELBASSA, Fleur BOURDELAIS,
Sébastien DURAND, Emilie THOMAS, Anthony FERRARI and Virginie
MARCEL</h4>
            
            <h4 data-toc-skip class="date">09/08/2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RibosomeCRCL/rRMSAnalyzer/blob/main/vignettes/rRMSAnalyzer.Rmd" class="external-link"><code>vignettes/rRMSAnalyzer.Rmd</code></a></small>
      <div class="d-none name"><code>rRMSAnalyzer.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>RiboMethSeq is an innovative RNAseq-based approach, which was
      developed in 2015 to analyze 2’O-ribose methylation (2’Ome) at all
      sites of ribosomal RNAs (rRNA) at once, in yeast <span class="citation">(Birkedal et al. 2014)</span>. This approach was
      then transferred to human using the Illumina technology <span class="citation">(Marchand et al. 2016; Erales et al. 2017; Marcel
      et al. 2021)</span>. Briefly, the presence of 2’Ome protects the
      phosphodiester bond located at the 3’ of the 2’Ome nucleotide from
      alkaline hydrolysis. Thus, the presence of 2’Ome at the given
      nucleotide n induces under-representation of RNA fragments
      starting at the nucleotide n+1 and an over-representation of RNA
      fragments ending at the nucleotide n-1, allowing the extrapolation
      of 2’Ome levels at the corresponding nucleotide position n (or
      C-score) varying from 0 to 1 <span class="citation">(Birkedal et
      al. 2014)</span>. <br><br> The rRMSAnalyzer package can be used
      for any kind of RNA and with all organisms. This package provides
      a set of user-friendly functions to compute C-scores from
      RiboMethSeq read end counts as input, adjust batch effect with
      ComBat-Seq, visualize the data and provide a table with the
      annotated human rRNA sites and their C-scores. This package
      provides a set of user-friendly functions to (i) compute C-scores
      from RiboMethSeq read-end counts as input,(ii) perform quality
      control of the dataset, (iii) adjust batch effect with ComBat-Seq,
      and (iv) provide tools to visualize, analyze the data and provide
      a table with the annotated human rRNA sites and their C-scores
      (Paraqindes et al., 2023), (v) return automated QC and analytic
      reports. Raw data processing to obtain read-end counts from
      sequencing data can be performed using our Nextflow pipeline <a href="https://github.com/RibosomeCRCL/ribomethseq-nf" class="external-link">ribomethseq-nf</a>,
      as already described <span class="citation">(Marchand et al.
      2016)</span>.</p>
    </div>
    
<div class="section level2">
<h2 id="general-information">General information<a class="anchor" aria-label="anchor" href="#general-information"></a>
</h2>
<div class="section level3">
<h3 id="help-bug-reports-and-suggestions">Help, bug reports and suggestions<a class="anchor" aria-label="anchor" href="#help-bug-reports-and-suggestions"></a>
</h3>
<p>To report a bug or any suggestion to improve the package, please let
us know by opening a new issue on: <a href="https://github.com/RibosomeCRCL/rRMSAnalyzer/issues" class="external-link uri">https://github.com/RibosomeCRCL/rRMSAnalyzer/issues</a></p>
</div>
<div class="section level3">
<h3 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h3>
<p>We would like to thank all our collaborators for their advice and
suggestions.</p>
</div>
<div class="section level3">
<h3 id="funding">Funding<a class="anchor" aria-label="anchor" href="#funding"></a>
</h3>
<p>This project has been funded by the French Cancer Institute (INCa,
PLBIO 2019-138 MARACAS), the SIRIC Program (INCa-DGOS-Inserm_12563
LyRICAN) and <a href="https://www.synergielyoncancer.fr" class="external-link">Synergie Lyon
Cancer Foundation</a>.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>The latest version of rRMSAnalyzer package can be installed from
Github with :</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"RibosomeCRCL/rRMSAnalyzer"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RibosomeCRCL/rRMSAnalyzer" class="external-link">rRMSAnalyzer</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="initial-preparation-of-the-dataset">Initial preparation of the dataset<a class="anchor" aria-label="anchor" href="#initial-preparation-of-the-dataset"></a>
</h2>
<div class="section level3">
<h3 id="riboclass">RiboClass<a class="anchor" aria-label="anchor" href="#riboclass"></a>
</h3>
<p>RiboClass is the main class of the package that enables the storage
of both the data matrices (read-end counts and C-scores) and the
associated metadata. It is automatically created when calling
<em><code><a href="../reference/load_ribodata.html">load_ribodata()</a></code></em> (see <a href="#loading-data">Loading data</a>).</p>
<p>It is a list containing three main elements, as described below:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Data</strong>: a <em>list of dataframe</em>, containing
for each sample the 5’ and/or 3’ read-end counts provided by the user,
and the calculated C-score.</p></li>
<li><p><strong>Metadata</strong>: a <em>dataframe</em>, containing all
the information related to the samples that can be provided by the
user.</p></li>
<li><p><strong>rRNA_names</strong>: a <em>dataframe</em>, reporting the
names of the rRNA used in Data.</p></li>
</ol>
<p>Some major function parameters (such as the normalization method used
for C-score computation) are also kept in the RiboClass object as a
reminder.</p>
</div>
<div class="section level3">
<h3 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<div class="section level4">
<h4 class="unnumbered" id="data-to-provide">Data to provide<a class="anchor" aria-label="anchor" href="#data-to-provide"></a>
</h4>
<div class="section level5">
<h5 class="unnumbered" id="read-end-counts">Read-end counts<a class="anchor" aria-label="anchor" href="#read-end-counts"></a>
</h5>
<p>To use this package, the user must provide at least one csv/tsv file
with the 5’, 3’ or 5’/3’ read-end counts resulting from RiboMethSeq data
per sample. The folder structure containing the csv files is not
important, as long as either the directory and its sub-directories
contain the necessary csv/tsv files.</p>
<ol style="list-style-type: decimal">
<li><p>The <strong>name of the rRNA</strong> on which the read end
counting was performed.</p></li>
<li><p>The <strong>position’s number</strong> on the rRNA.</p></li>
<li><p>The <strong>value of the read-end counts</strong> at the
position.</p></li>
</ol>
<p>Here is an example :</p>
<table class="table">
<thead><tr class="header">
<th>rRNA</th>
<th>Position on rRNA</th>
<th>read end count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>18S</td>
<td>123</td>
<td>3746</td>
</tr>
<tr class="even">
<td>18S</td>
<td>124</td>
<td>345</td>
</tr>
<tr class="odd">
<td>18S</td>
<td>125</td>
<td>324</td>
</tr>
<tr class="even">
<td>18S</td>
<td>126</td>
<td>789</td>
</tr>
<tr class="odd">
<td>18S</td>
<td>127</td>
<td>1234</td>
</tr>
</tbody>
</table>
<p>Note 1: it is not necessary to provide a header in the count files,
because column index can be used in the function
<em>load_ribodata()</em>, using <em><code>count_value()</code></em>,
<em><code>count_rnaid()</code></em> and
<em><code>count_pos()</code></em> arguments.</p>
<p>Note 2: if no metadata is specified (metadata = NULL), rRMSAnalyzer
will try to fetch any csv files in the folder specified in count_path
and its subfolders.</p>
</div>
<div class="section level5">
<h5 class="unnumbered" id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h5>
<p>The expected metadata is either a dataframe already in the R
environment or a csv/tsv file.</p>
<p><u>Two columns are mandatory for the metadata :</u></p>
<ol style="list-style-type: decimal">
<li><p><strong>filename</strong>: name of the csv file on disk
containing the read-end counts described above. Do not modify it unless
the filename has changed on disk.</p></li>
<li><p><strong>samplename</strong>: rename the samples that will be
analyzed and displayed on the plots. This column can be modified, as
long as the sample names are unique.</p></li>
</ol>
<p>After these two mandatory columns, the user can provide as many
columns as needed for the analysis.</p>
<p>Here is an example of metadata for 3 samples:</p>
<table style="width:74%;" class="table">
<colgroup>
<col width="20%">
<col width="20%">
<col width="31%">
</colgroup>
<thead><tr class="header">
<th>
<p>filename</p>
<p>(mandatory)</p>
</th>
<th>
<p>samplename</p>
<p>(mandatory)</p>
</th>
<th>
<p>condition</p>
<p>(optional) |</p>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample1.csv</td>
<td>sample 1</td>
<td>condition 1</td>
</tr>
<tr class="even">
<td>sample2.csv</td>
<td>sample 2</td>
<td>condition 1</td>
</tr>
<tr class="odd">
<td>sample3.csv</td>
<td>sample 3</td>
<td>condition 2</td>
</tr>
</tbody>
</table>
<p>Note: if no metadata is provided in
<em><code><a href="../reference/load_ribodata.html">load_ribodata()</a></code></em> (metadata = NULL), an empty
metadata will be created with the “filename” and “samplename” columns
pre-filled. The “samplename” column will be identical to “filename”, but
can be modified by the user.</p>
<p>Here is an example of auto-generated metadata:</p>
<table class="table">
<thead><tr class="header">
<th>filename</th>
<th>samplename</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample1.csv</td>
<td>sample1.csv</td>
</tr>
<tr class="even">
<td>sample2.csv</td>
<td>sample2.csv</td>
</tr>
<tr class="odd">
<td>sample3.csv</td>
<td>sample3.csv</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 class="unnumbered" id="how-to-load-the-data">How to load the data<a class="anchor" aria-label="anchor" href="#how-to-load-the-data"></a>
</h5>
<p>To load both data and metadata, and store them in a RiboClass, the
function <em><code><a href="../reference/load_ribodata.html">load_ribodata()</a></code></em> is used.</p>
<p>The following example displays a call to
<em><code><a href="../reference/load_ribodata.html">load_ribodata()</a></code></em>, with all arguments shown :</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"rRMSAnalyzer"</span><span class="op">)</span> <span class="co">#change for your path</span></span>
<span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_ribodata.html">load_ribodata</a></span><span class="op">(</span></span>
<span>                      <span class="co">#data &amp; metadata files path</span></span>
<span>                      count_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">path</span>,<span class="st">"miniglioma/"</span><span class="op">)</span>,</span>
<span>                      metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">path</span>,<span class="st">"metadata.csv"</span><span class="op">)</span>,</span>
<span>                      <span class="co"># data &amp; metadata files separator</span></span>
<span>                      count_sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>                      metadata_sep <span class="op">=</span> <span class="st">","</span>,</span>
<span>                      <span class="co"># count data arguments :</span></span>
<span>                      count_header <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                      count_value <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                      count_rnaid <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                      count_pos <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                      <span class="co"># Metadata arguments :</span></span>
<span>                      metadata_key <span class="op">=</span> <span class="st">"filename"</span>,</span>
<span>                      metadata_id <span class="op">=</span> <span class="st">"samplename"</span>,</span>
<span>                      <span class="co"># C-score arguments :</span></span>
<span>                      flanking <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>                      ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Your data has been successfully loaded!</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level3">
<h3 class="unnumbered" id="rrna-names">rRNA names<a class="anchor" aria-label="anchor" href="#rrna-names"></a>
</h3>
<div class="section level4">
<h4 class="unnumbered" id="provided">Provided<a class="anchor" aria-label="anchor" href="#provided"></a>
</h4>
<p>rRNA names are automatically obtained from the data and stored in a
generated dataframe inside the RiboClass. It contains two columns :</p>
<ol style="list-style-type: decimal">
<li><p><strong>original_name</strong>: original name of each rRNA (e.g
NR_023363.1).</p></li>
<li><p><strong>current_name</strong>: current name of each rRNA,
reflecting any user’s change with <em><code><a href="../reference/rename_rna.html">rename_rna()</a></code></em>
function (see [Rename RNA]).</p></li>
</ol>
<p>This dataframe is used to keep track of the original name, which
often includes the NCBI’s accession ID.</p>
<p>Here is an example:</p>
<table class="table">
<thead><tr class="header">
<th>original_name</th>
<th>current_name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>NR_023363.1_5S</td>
<td>5S</td>
</tr>
<tr class="even">
<td>NR_046235.3_5.8S</td>
<td>5.8S</td>
</tr>
<tr class="odd">
<td>NR_046235.3_18S</td>
<td>18S</td>
</tr>
<tr class="even">
<td>NR_046235.3_28S</td>
<td>28S</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 class="unnumbered" id="custom">Custom<a class="anchor" aria-label="anchor" href="#custom"></a>
</h4>
<p>The user must not modify this dataframe manually. To rename or remove
rRNA, the user can use <em><code><a href="../reference/rename_rna.html">rename_rna()</a></code></em> and
<em><code><a href="../reference/remove_rna.html">remove_rna()</a></code></em>. The dataframe will be updated
accordingly.</p>
</div>
</div>
<div class="section level3">
<h3 id="c-score-calculation">C-score calculation<a class="anchor" aria-label="anchor" href="#c-score-calculation"></a>
</h3>
<div class="section level4">
<h4 class="unnumbered" id="what-is-a-c-score">What is a C-score ?<a class="anchor" aria-label="anchor" href="#what-is-a-c-score"></a>
</h4>
<p>The C-score is a metric, calculated from the RiboMethSeq sequencing
data, used to evaluate the level of 2’O-ribose methylation (2’Ome) at a
given position in the rRNA.</p>
<p>The C-score is calculated by normalizing the end read count with
respect to the local environment at each genomic position and directly
indicates the rRNA 2’Ome level. The C-score ranges from 0 (i.e., no rRNA
molecules of the sample are 2’Ome at this specific site) to 1 (i.e., all
rRNA molecules of the sample are 2’Ome at this specific site). A C-score
with an intermediate value between 0 to 1 means that the sample contains
a mixture of un-2’Ome and 2’Ome rRNA molecules.</p>
<p>To obtain a robust estimate of the 2’Ome level, different C-scores
can be determined depending on the arguments used to compute the local
coverage. In particular, the estimation method and the size of the local
coverage to be considered can be changed.</p>
<p>By default, the local coverage is estimated by calculating the
<strong>median</strong> of the <strong>5’ read-end counts</strong> in a
<strong>flanking region of 6</strong> (i.e., 6 nucleotides downstream
the nucleotide n and 6 upstream the nucleotide n, where n is the
nucleotid directly following the 2’Ome site of interest). This package
provides the ability to change these two arguments either when loading
the data or during the analysis.</p>
<div class="section level5">
<h5 class="unnumbered" id="c-score-computation-when-loading-data">C-score computation when loading data<a class="anchor" aria-label="anchor" href="#c-score-computation-when-loading-data"></a>
</h5>
<p>When using the <em><code><a href="../reference/load_ribodata.html">load_ribodata()</a></code></em> function, a
C-score is automatically calculated for all genomic positions of the
rRNA. The C-score is computed using either the default arguments of the
<em><code><a href="../reference/load_ribodata.html">load_ribodata()</a></code></em> function or user-defined arguments
as follows:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/load_ribodata.html">load_ribodata</a></span><span class="op">(</span>count.path <span class="op">=</span> <span class="st">"/path/to/csv/"</span>,</span>
<span>              metadata <span class="op">=</span> <span class="st">"/path/to/metadata.csv"</span>,</span>
<span>              <span class="co"># everything below is linked to C-score computation</span></span>
<span>              flanking <span class="op">=</span> <span class="fl">6</span>, <span class="co"># flanking region size</span></span>
<span>              method <span class="op">=</span> <span class="st">"median"</span>, <span class="co"># use mean or median on flanking region's values</span></span>
<span>              ncores <span class="op">=</span> <span class="fl">8</span> <span class="co"># number of CPU cores to use for computation</span></span>
<span>              <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 class="unnumbered" id="c-score-computation-during-the-analysis">C-score computation during the analysis<a class="anchor" aria-label="anchor" href="#c-score-computation-during-the-analysis"></a>
</h5>
<p>During the analysis, the C-score calculating arguments (method and
size of the flanking region) can be modified using the
<em><code><a href="../reference/compute_cscore.html">compute_cscore()</a></code></em> function, which will
automatically update the C-score in the RiboClass.</p>
<p>In the following example, both the flanking region’s size of the
local coverage and the computation method have been modified:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compute the C-score using different arguments,</span></span>
<span><span class="co"># including calculation of the local coverage using the mean instead of the median</span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_cscore.html">compute_cscore</a></span><span class="op">(</span><span class="va">ribo</span>,</span>
<span>                       flanking <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                       method <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span></span></code></pre></div>
<p><strong>Important: this function will override the previous C-score
of the RiboClass.</strong></p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="quality-control">Quality control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<p>Due to technical limitations, it is sometimes necessary to conduct
wet-lab preparations of large cohorts in several batches. The main risk
when making several batches is to introduce <strong>technical
biases</strong> or <strong>batch effect</strong> in the dataset.</p>
<div class="section level3">
<h3 id="content-of-the-qc-report">Content of the QC report<a class="anchor" aria-label="anchor" href="#content-of-the-qc-report"></a>
</h3>
<p>A quality control (QC) report can be performed. It uses several
metrics to help identify outlier samples and/or batch effects, including
read-end counts and the C-score itself at all the 7217 genomic
positions. The QC verifies that the coverage is uniform and reproducible
between samples, eliminating the possibility of bias due to sequencing
and outliers.</p>
<p>QC can be either performed using a panel of ready to use functions,
which correspond to data visualization, or automatically.</p>
<p>The automatic QC report can be generated using the
<em><code><a href="../reference/report_qc.html">report_qc()</a></code></em> function:</p>
<p>With the ribo_toy example, the column that contains the information
about samples batches is called “run”. The name is thus given to the
<em><code>library_col</code></em> argument but it can be also left
empty.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/report_qc.html">report_qc</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo_toy</span>, library_col <span class="op">=</span> <span class="st">"run"</span><span class="op">)</span></span></code></pre></div>
<p>The QC report includes the following visualizations:</p>
<ul>
<li>Read-end counts distribution by sample
(<em><code><a href="../reference/boxplot_count.html">boxplot_count()</a></code></em>)</li>
<li>Relative log coverage (i.e., end read count) by sample
(<em><code><a href="../reference/plot_rlc.html">plot_rlc()</a></code></em>)</li>
<li>rRNA fraction of read-end counts per sample
(<em><code>plot_count_fraction()</code></em>)</li>
<li>Heatmap summarizing the correlation matrix of the read-end counts
(<em><code>heatmap_annotated()</code></em>)</li>
<li>Correspondence analysis of the read-end counts
(<em><code><a href="../reference/plot_coa.html">plot_coa()</a></code></em>)</li>
<li>Principal component analysis (<em><code><a href="../reference/plot_pca.html">plot_pca()</a></code></em>)</li>
</ul>
<div class="section level4">
<h4 class="unnumbered" id="correlation-matrix">Correlation matrix<a class="anchor" aria-label="anchor" href="#correlation-matrix"></a>
</h4>
<p>To add outlier annotations on the heatmap,
<em><code><a href="../reference/get_outliers.html">get_outliers()</a></code></em> function must be used. This
function computes all the outliers identified in the QC and returns a
data frame summary for use in
<em><code><a href="../reference/plot_heatmap_annotated.html">plot_heatmap_annotated()</a></code></em></p>
<p>Here is an example of correlation matrix with outlier
identification:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qcdata</span> <span class="op">&lt;-</span> <span class="fu">rRMSAnalyzer</span><span class="fu">::</span><span class="fu"><a href="../reference/get_outliers.html">get_outliers</a></span><span class="op">(</span><span class="va">ribo</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_heatmap_annotated.html">plot_heatmap_annotated</a></span><span class="op">(</span><span class="va">ribo</span>, <span class="va">qcdata</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-7-1.png" width="672"></p>
</div>
<div class="section level4">
<h4 class="unnumbered" id="output-of-the-qc-report">Output of the QC report<a class="anchor" aria-label="anchor" href="#output-of-the-qc-report"></a>
</h4>
<p>Information issued from the QC report can be return as followed:</p>
<p>Here is an example of QC output:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># To display the output</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">qcdata</span>, <span class="st">"html"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">kable_styling</span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">scroll_box</span><span class="op">(</span>width <span class="op">=</span> <span class="st">"100%"</span>, height <span class="op">=</span> <span class="st">"500px"</span><span class="op">)</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:100%; ">
<table class="table table table-striped" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
samplename
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
condition
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
run
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
filename
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
median_coverage
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
coverage_quality
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
rlc_median
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
rlc_median_quality
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
total_outliers
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
cond1
</td>
<td style="text-align:left;">
L2
</td>
<td style="text-align:left;">
S1_5_counts.csv
</td>
<td style="text-align:right;">
1761
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
0.0575140
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S2
</td>
<td style="text-align:left;">
S2
</td>
<td style="text-align:left;">
cond2
</td>
<td style="text-align:left;">
L1
</td>
<td style="text-align:left;">
S2_5_counts.csv
</td>
<td style="text-align:right;">
1292
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
-0.3439012
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S3
</td>
<td style="text-align:left;">
S3
</td>
<td style="text-align:left;">
cond2
</td>
<td style="text-align:left;">
L2
</td>
<td style="text-align:left;">
S3_5_counts.csv
</td>
<td style="text-align:right;">
2212
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
0.3954259
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S4
</td>
<td style="text-align:left;">
S4
</td>
<td style="text-align:left;">
cond2
</td>
<td style="text-align:left;">
L1
</td>
<td style="text-align:left;">
S4_5_counts.csv
</td>
<td style="text-align:right;">
2147
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
0.3337012
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S5
</td>
<td style="text-align:left;">
S5
</td>
<td style="text-align:left;">
cond2
</td>
<td style="text-align:left;">
L2
</td>
<td style="text-align:left;">
S5_5_counts.csv
</td>
<td style="text-align:right;">
1682
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
-0.0113976
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S6
</td>
<td style="text-align:left;">
S6
</td>
<td style="text-align:left;">
cond2
</td>
<td style="text-align:left;">
L2
</td>
<td style="text-align:left;">
S6_5_counts.csv
</td>
<td style="text-align:right;">
1774
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
0.0499721
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S7
</td>
<td style="text-align:left;">
S7
</td>
<td style="text-align:left;">
cond2
</td>
<td style="text-align:left;">
L1
</td>
<td style="text-align:left;">
S7_5_counts.csv
</td>
<td style="text-align:right;">
696
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
-1.2547609
</td>
<td style="text-align:left;">
warning
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S8
</td>
<td style="text-align:left;">
S8
</td>
<td style="text-align:left;">
cond1
</td>
<td style="text-align:left;">
L1
</td>
<td style="text-align:left;">
S8_5_counts.csv
</td>
<td style="text-align:right;">
1759
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
0.0618611
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S9
</td>
<td style="text-align:left;">
S9
</td>
<td style="text-align:left;">
cond1
</td>
<td style="text-align:left;">
L2
</td>
<td style="text-align:left;">
S9_5_counts.csv
</td>
<td style="text-align:right;">
1394
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
-0.2504866
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S10
</td>
<td style="text-align:left;">
S10
</td>
<td style="text-align:left;">
cond1
</td>
<td style="text-align:left;">
L1
</td>
<td style="text-align:left;">
S10_5_counts.csv
</td>
<td style="text-align:right;">
1625
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
-0.0767012
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S11
</td>
<td style="text-align:left;">
S11
</td>
<td style="text-align:left;">
cond1
</td>
<td style="text-align:left;">
L1
</td>
<td style="text-align:left;">
S11_5_counts.csv
</td>
<td style="text-align:right;">
1632
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
-0.0471291
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S12
</td>
<td style="text-align:left;">
S12
</td>
<td style="text-align:left;">
cond1
</td>
<td style="text-align:left;">
L2
</td>
<td style="text-align:left;">
S12_5_counts.csv
</td>
<td style="text-align:right;">
1683
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
0.0062753
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
RNA1
</td>
<td style="text-align:left;">
RNA1
</td>
<td style="text-align:left;">
RNA ref
</td>
<td style="text-align:left;">
L1
</td>
<td style="text-align:left;">
RNA1_5_counts.csv
</td>
<td style="text-align:right;">
1593
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
-0.1061994
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
RNA2
</td>
<td style="text-align:left;">
RNA2
</td>
<td style="text-align:left;">
RNA ref
</td>
<td style="text-align:left;">
L2
</td>
<td style="text-align:left;">
RNA2_5_counts.csv
</td>
<td style="text-align:right;">
2316
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:right;">
0.3822330
</td>
<td style="text-align:left;">
pass
</td>
<td style="text-align:left;">
0
</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To save the output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">qcdata</span>,<span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">output_dir</span>,<span class="st">"quality_control.csv"</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"\t"</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 class="unnumbered" id="comments">Comments<a class="anchor" aria-label="anchor" href="#comments"></a>
</h4>
<p>A Comments section is available in the QC report to add some personal
comments. To do so, a Rmarkdown file should be created and saved in the
working directory. Then, path to this report should be given in the
<em>comments</em> argument of the <em><code><a href="../reference/report_qc.html">report_qc()</a></code></em>
function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/report_qc.html">report_qc</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo</span>, specie <span class="op">=</span> <span class="st">"human"</span>, library_col <span class="op">=</span> <span class="st">"run"</span>, comments <span class="op">=</span> <span class="st">"./path/to/comment_QC.Rmd"</span><span class="op">)</span></span></code></pre></div>
<p>Example of comments in rmd extension</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="st">```</span><span class="at">{=html} </span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="at">&lt;style type="text/css"&gt;</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="at">  body {</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="at">    font-family: Helvetica;</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="at">  }</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="at">  /* Apply a skyblue background to the blocks &lt;details&gt; */</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="at">  details {</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="at">    background-color: #9ecae1;</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="at">    padding: 10px;</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="at">    border-radius: 5px;</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="at">  }</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="at">&lt;/style&gt;</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="st">```</span> <span class="co"># erase this '#' and this sentence</span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="sc">&lt;</span>details<span class="sc">&gt;</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a><span class="er">&lt;</span>summary<span class="sc">&gt;</span><span class="er">&lt;</span>b<span class="sc">&gt;</span> ℹ Comments on the report  <span class="sc">&lt;</span><span class="er">/</span>b<span class="sc">&gt;</span><span class="er">&lt;/</span>summary<span class="sc">&gt;</span></span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="er">&lt;</span>div markdown<span class="ot">=</span><span class="st">"1"</span><span class="sc">&gt;</span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="er">&lt;</span>p<span class="sc">&gt;</span></span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a>Text can be entered from here ...</span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a>Subtitle can be added with <span class="st">'##'</span> symbol<span class="sc">:</span></span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a></span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a><span class="do">## 5.3 Level 2 title {-} #this symbol {-} avoid continue the numbering of the table of content</span></span>
<span id="cb10-25"><a href="#cb10-25" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" tabindex="-1"></a>...to here. Those last <span class="dv">2</span> lines below must be kept. They are usefull <span class="cf">for</span> the layout.</span>
<span id="cb10-27"><a href="#cb10-27" tabindex="-1"></a><span class="sc">&lt;</span><span class="er">/</span>div<span class="sc">&gt;</span></span>
<span id="cb10-28"><a href="#cb10-28" tabindex="-1"></a><span class="er">&lt;/</span>details<span class="sc">&gt;</span></span>
<span id="cb10-29"><a href="#cb10-29" tabindex="-1"></a>  </span>
<span id="cb10-30"><a href="#cb10-30" tabindex="-1"></a><span class="co">#copy/past in an rmd file</span></span>
<span id="cb10-31"><a href="#cb10-31" tabindex="-1"></a><span class="co">#apply in the report_qc() function</span></span>
<span id="cb10-32"><a href="#cb10-32" tabindex="-1"></a><span class="co">#report_qc(ribo = ribo….)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="batch-effect">Batch effect<a class="anchor" aria-label="anchor" href="#batch-effect"></a>
</h3>
<div class="section level4">
<h4 class="unnumbered" id="batch-effect-identification">Batch effect identification<a class="anchor" aria-label="anchor" href="#batch-effect-identification"></a>
</h4>
<p>Technical bias (i.e., batch effect) can be identified by plotting
C-scores at all the genomic positions of the rRNA for each sample on a
PCA (see also [Visualization with PCA] for more uses).</p>
<p>Since more than 1.5% of the C-score corresponds to non-biological
noise, normalization during C-score calculation should limit the
dispersion of the samples based on their C-score at all genomic
positions in the whole series. Thus, Principal Component Analysis (PCA)
on C-scores helps to identify putative batch effects that can be
corrected using the inter-normalization option (ComBat-seq tool) (Zhang
et al, 2020; Paraqindes et al, 2023). <br><br> The graph shows a
Principal Component Analysis (PCA) plot illustrating the distance
between each sample based on its C-score at all genomic positions. The
samples are colored by library.</p>
<p>Here is an example:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot a Principal Component Analysis (PCA) whose colors depend on the "run" column in metadata</span></span>
<span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo</span>,</span>
<span>         color_col <span class="op">=</span> <span class="st">"run"</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>In this example, the technical replicates RNA1 and RNA2 included in
library 1 and 2 respectively, are distant from each other on the PC1
axis. Moreover, the samples should not be grouped by library or batch.
The following section will resolve this batch effect.</p>
</div>
<div class="section level4">
<h4 class="unnumbered" id="batch-effect-adjustment">Batch effect adjustment<a class="anchor" aria-label="anchor" href="#batch-effect-adjustment"></a>
</h4>
<p>Batch effect of RiboMethSeq data can be adjusted using the ComBat-seq
method (Paraqindes et al, 2023 ; Zhang, et al, 2020). The rRMSAnalyzer
package includes a wrapper (adjust_bias) to perform ComBat-seq
adjustment which provides a new RiboClass with adjusted read end count
values and C-scores automatically recalculated with the same setup
arguments.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ribo_toy"</span><span class="op">)</span></span>
<span><span class="co"># If necessary, adjust any technical biases using ComBat-Seq.</span></span>
<span><span class="co"># Here, as an example, we use the "run" column in metadata.</span></span>
<span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/adjust_bias.html">adjust_bias</a></span><span class="op">(</span><span class="va">ribo_toy</span>, batch <span class="op">=</span> <span class="st">"run"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Found 2 batches</span></span>
<span><span class="co">#&gt; Using null model in ComBat-seq.</span></span>
<span><span class="co">#&gt; Adjusting for 0 covariate(s) or covariate level(s)</span></span>
<span><span class="co">#&gt; Estimating dispersions</span></span>
<span><span class="co">#&gt; Fitting the GLM model</span></span>
<span><span class="co">#&gt; Shrinkage off - using GLM estimates for parameters</span></span>
<span><span class="co">#&gt; Adjusting the data</span></span>
<span><span class="co">#&gt; Recomputing c-score with the following parameters :</span></span>
<span><span class="co">#&gt; - C-score method : median</span></span>
<span><span class="co">#&gt; - Flanking window : 6</span></span></code></pre></div>
<p>Batch effect adjustment can be verified using the plot_pca function
on the new RiboClass:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="st">"run"</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>After batch effect adjustment using ComBat-seq method, the two
technical replicates RNA1 and RNA2 show reduced dispersion, and the
samples are separated on the PCA axes independently of the library they
belong to.</p>
</div>
</div>
<div class="section level3">
<h3 id="verifying-local-coverage">Verifying local coverage<a class="anchor" aria-label="anchor" href="#verifying-local-coverage"></a>
</h3>
<p>The C-score depends on the local coverage around the site. For a
simple visualization of the read end count coverage of the flanking
region of a given genomic position of interest, use the
<code><a href="../reference/plot_counts_env.html">plot_counts_env()</a></code> function. Two plotting methods are
available:</p>
<ol style="list-style-type: decimal">
<li>
<strong>display all samples</strong> (default arguments): by
displaying the end read count coverage of all samples using a boxplot at
the +/-6 genomic position relative to the genomic position of interest
(green box). A pink box is displayed if there is other modification in
the window. Such a plot is automatically used when samples = “all”. The
median of the read end count and coverage limit are shown as two dashed
red lines.</li>
</ol>
<p>Here is an example:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_counts_env.html">plot_counts_env</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="st">"NR_046235.3_5.8S"</span>,<span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<ol start="2" style="list-style-type: decimal">
<li>
<strong>sample of interest</strong>: by displaying the end read
count coverage of the sample of interest only using the profile at the
+/-6 genomic position relative to the genomic position of interest
(vertical green line). The median of the read end count is displayed as
a horizontal dotted red line.</li>
</ol>
<p>Here is an example:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_counts_env.html">plot_counts_env</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="st">"NR_046235.3_5.8S"</span>,<span class="fl">14</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S1"</span>,<span class="st">"S2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="data-formatting-for-subsequent-analyses">Data formatting for subsequent analyses<a class="anchor" aria-label="anchor" href="#data-formatting-for-subsequent-analyses"></a>
</h2>
<div class="section level3">
<h3 id="sample-manipulation">Sample manipulation<a class="anchor" aria-label="anchor" href="#sample-manipulation"></a>
</h3>
<div class="section level4">
<h4 class="unnumbered" id="keep-or-remove-samples">Keep or remove samples<a class="anchor" aria-label="anchor" href="#keep-or-remove-samples"></a>
</h4>
<p>A sample subset can be easily analyzed by specifying which samples to
keep or which to remove. The user can then create a new RiboClass object
containing the data and metadata of the samples of interest. In both
cases, only the metadata of the remaining samples are kept in the
RiboClass object, so no manual updating is required.</p>
<p>Here is an example of how to create a new RiboClass by retaining two
samples of interest (“S1” and “S2”):</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_2samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/keep_ribo_samples.html">keep_ribo_samples</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S1"</span>,<span class="st">"S2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ribo_2samples</span><span class="op">)</span></span>
<span><span class="co">#&gt; a RiboClass with 2 samples and 4 RNA(s) :</span></span>
<span><span class="co">#&gt; Name : NR_023363.1_5S, length : 121</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_5.8S, length : 157</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_18S, length : 1869</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_28S, length : 5070</span></span></code></pre></div>
<p>Here is an example to generate a new RiboClass by removing two
samples (“RNA1” and “RNA2”):</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_ribo_samples.html">remove_ribo_samples</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RNA1"</span>,<span class="st">"RNA2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ribo_adjusted</span><span class="op">)</span></span>
<span><span class="co">#&gt; a RiboClass with 12 samples and 4 RNA(s) :</span></span>
<span><span class="co">#&gt; Name : NR_023363.1_5S, length : 121</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_5.8S, length : 157</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_18S, length : 1869</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_28S, length : 5070</span></span></code></pre></div>
<p>In both cases, only the remaining samples’ metadata are kept in the
RiboClass object. There is no need to update it manually.</p>
</div>
</div>
<div class="section level3">
<h3 id="rrna-manipulation-and-annotation">rRNA manipulation and annotation<a class="anchor" aria-label="anchor" href="#rrna-manipulation-and-annotation"></a>
</h3>
<div class="section level4">
<h4 class="unnumbered" id="rrna-manipulation">rRNA manipulation<a class="anchor" aria-label="anchor" href="#rrna-manipulation"></a>
</h4>
<div class="section level5">
<h5 class="unnumbered" id="remove-rrna">Remove rRNA<a class="anchor" aria-label="anchor" href="#remove-rrna"></a>
</h5>
<p>A subset of rRNA can be easily analyzed by specifying the rRNA to be
removed. The user can thus create a new RiboClass object containing the
data of the rRNAs of interest, without affecting the metadata of the
samples.</p>
<p>Here is an example where the rRNA 5S is removed:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_rna.html">remove_rna</a></span><span class="op">(</span><span class="va">ribo</span>, rna_to_remove <span class="op">=</span> <span class="st">"NR_023363.1_5S"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ribo_adjusted</span><span class="op">)</span></span>
<span><span class="co">#&gt; a RiboClass with 14 samples and 3 RNA(s) :</span></span>
<span><span class="co">#&gt; Name : NR_046235.3_5.8S, length : 157</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_18S, length : 1869</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_28S, length : 5070</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="rename-rrna">Rename rRNA<a class="anchor" aria-label="anchor" href="#rename-rrna"></a>
</h4>
<p>The annotation of rRNA 2’Ome sites using the lists provided by this
package requires the use of specific rRNA names.</p>
<p>Here is an example to check if the rRNA names provided by the user in
the RiboClass match the ones used by this package :</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"human_methylated"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"human_methylated's rna names: "</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">human_methylated</span><span class="op">$</span><span class="va">rRNA</span><span class="op">)</span>,<span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; human_methylated's rna names:  5.8S 18S 28S</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"ribo's rna names: "</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">ribo_adjusted</span><span class="op">$</span><span class="va">rna_names</span><span class="op">$</span><span class="va">current_name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; ribo's rna names:  NR_046235.3_5.8S NR_046235.3_18S NR_046235.3_28S</span></span></code></pre></div>
<p>In this example, the names are different and need to be updated
before annotation.</p>
<p>The <code><a href="../reference/rename_rna.html">rename_rna()</a></code> function automatically updates the rRNA
names given by the rRNA size order:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rename_rna.html">rename_rna</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                            new_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"5.8S"</span>, <span class="st">"18S"</span>, <span class="st">"28S"</span><span class="op">)</span><span class="op">)</span> </span>
<span>                            <span class="co"># from the shortest rRNA in our RiboClass to the longest.</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="annotation-of-rrna-2ome-sites">Annotation of rRNA 2’Ome sites<a class="anchor" aria-label="anchor" href="#annotation-of-rrna-2ome-sites"></a>
</h3>
<p>The rRMSAnalyzer package calculates a C-score for each genomic
position of the rRNAs. However, not every site of the reference 2’Ome
rRNA is necessarily methylated. Therefore, it is expected that the user
provides a list of potentially methylated sites of interest, called
“annotated sites”. This will make it easier to restrict further analysis
to this list of sites. By default, rRMSAnalyzer package includes its own
human annotated sites.</p>
<div class="section level4">
<h4 class="unnumbered" id="included-annotation-human-2ome-rrna-sites">Included annotation : Human 2’Ome rRNA sites<a class="anchor" aria-label="anchor" href="#included-annotation-human-2ome-rrna-sites"></a>
</h4>
<p>By default, rRMSAnalyzer package includes three dataframes containing
the positions and the annotations of the human rRNA 2’Ome sites:</p>
<ul>
<li><p>human_methylated: a dataframe, containing the 112 known 2’Ome
sites for the human rRNAs.</p></li>
<li><p>human_suspected: a dataframe, containing the 17 sites that are
putative 2’Ome sites for the human rRNAs, as described in the
litterature.</p></li>
<li><p>human_both_methylated_and_suspected: a dataframe, containing the
129 sites of both known 2’Ome sites and sites that are putative 2’Ome
sites for the human rRNAs.</p></li>
</ul>
</div>
<div class="section level4">
<h4 class="unnumbered" id="customize-2ome-sites-annotations">Customize 2’Ome sites annotations<a class="anchor" aria-label="anchor" href="#customize-2ome-sites-annotations"></a>
</h4>
<p>Instead of using the list of human rRNA 2’Ome sites provided by the
rRMSAnalyzer package, the user can provide their own list of annotated
sites which will be attached to the RiboClass object using the
<code><a href="../reference/annotate_site.html">annotate_site()</a></code> function (see Section : annotation of rRNA
2’Ome sites).</p>
<p>This <code><a href="../reference/annotate_site.html">annotate_site()</a></code> function expects an “annot”
argument which must be a dataframe object that contains the following
three mandatory columns :</p>
<ul>
<li><p><strong>RNA name</strong>: the name of the rRNA, matching the
rRNA name of the RiboClass.</p></li>
<li><p><strong>Position on RNA</strong>: the number of the position on
the rRNA.</p></li>
<li><p><strong>Nomenclature</strong>: the name given to the site of
interest.</p></li>
</ul>
<p>You can see an example below :</p>
<table class="table">
<thead><tr class="header">
<th>Position</th>
<th>rRNA</th>
<th>Nomenclature</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>15</td>
<td>5.8S</td>
<td>Um14</td>
</tr>
<tr class="even">
<td>76</td>
<td>5.8S</td>
<td>Gm75</td>
</tr>
<tr class="odd">
<td>28</td>
<td>18S</td>
<td>Am27</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 class="unnumbered" id="annotate-rrna-sites">Annotate rRNA sites<a class="anchor" aria-label="anchor" href="#annotate-rrna-sites"></a>
</h4>
<p>The 2’Ome sites of interest must be attached to the RiboClass object
for further analysis using the <code><a href="../reference/annotate_site.html">annotate_site()</a></code> function
with either the provided annotations or custom annotations (see <a href="#customize-2ome-sites-annotations">Customize 2’Ome sites
annotations</a>).</p>
<p>Here is an example using the included human methylated
annotations:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_site.html">annotate_site</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                               annot <span class="op">=</span> <span class="va">human_methylated</span>,</span>
<span>                               anno_rna <span class="op">=</span> <span class="st">"rRNA"</span>,</span>
<span>                               anno_pos <span class="op">=</span> <span class="st">"Position"</span>,</span>
<span>                               anno_value <span class="op">=</span> <span class="st">"Nomenclature"</span><span class="op">)</span></span></code></pre></div>
<p>Note : rRMSAnalyzer will display a warning message when there is a
mismatch between the annot and RiboClass rRNA names.</p>
<p>This vignette also provides some explanations on how to create your
own sites annotation dataset with <a href="#customize-2ome-sites-annotations">Customize 2’Ome sites
annotations</a>.</p>
</div>
</div>
<div class="section level3">
<h3 id="analysis-design-for-multiple-comparisons">Analysis design for multiple comparisons<a class="anchor" aria-label="anchor" href="#analysis-design-for-multiple-comparisons"></a>
</h3>
<p>Although comparison of two groups can be computed using the analytic
report functions (e.g., report_2ome_sites or report_diff_sites – (see
[2’Ome profiling report] or <a href="#ome-site-by-site-report">2’Ome
site-by-site report</a>)) thanks to the “condition” column of the
metadata, this section needs to be considered to perform multiple
comparisons (i.e., more than 2 groups). The following section Addition
of new columns in the metadata is required for the two reports while the
section Comparison table is required for report_diff_sites only.</p>
<div class="section level4">
<h4 class="unnumbered" id="addition-of-new-columns-in-the-metadata">Addition of new columns in the metadata<a class="anchor" aria-label="anchor" href="#addition-of-new-columns-in-the-metadata"></a>
</h4>
<p>To perform multiple group comparison, the metadata needs to be
implemented with additional information on the experimental conditions
in order to create as much as novel Riboclass required to perform all
the comparison. The metadata are stored in the RiboClass object once it
has been created. To change the metadata, (i) amend the metadata
csv.file used to create the RiboClass (see <a href="#how-to-load-the-data">How to load the data</a>) and (ii) recreate
the RiboClass using the <em><code><a href="../reference/load_ribodata.html">load_ribodata()</a></code></em>
function.</p>
<p>Here is an example:</p>
<p>The study includes 3 experimental conditions termed “variable A to C”
(e.g., wild-type, mutation 1, mutation 2) that has been exposed to
another condition termed “ctrl, P1 and P2” (e.g., untreated vs treatment
with 2 different concentrations). All these experimental conditions have
been performed in replicates termed “N”. In summary, it includes:</p>
<ul>
<li>VariableA (9 samples) has 1 control (ctrl) and 2 cases (P1 and P2)
with 3 replicates for each (total = 9 samples). <br>
</li>
<li>VariableB (4 samples) has 1 control (ctrl) and 1 case (P2) with 2
replicates for each (total = 4 samples). <br>
</li>
<li>VariableC (6 samples) has 1 control (ctrl) and 1 case (P2) with 3
replicates for each (total = 6 samples). <br>
</li>
</ul>
</div>
<div class="section level4">
<h4 class="unnumbered" id="i-update-of-the-metadata--csv-file">(i) Update of the metadata .csv file<a class="anchor" aria-label="anchor" href="#i-update-of-the-metadata--csv-file"></a>
</h4>
<p>-&gt;&gt;&gt; Here is the initial metadata in the RiboClass:</p>
<table style="width:96%;" class="table">
<colgroup>
<col width="29%">
<col width="28%">
<col width="17%">
<col width="14%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th>filename</th>
<th>samplename</th>
<th>condition</th>
<th>treatment</th>
<th>lib</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>S26_R1.5_counts.csv</td>
<td>variableB_N2_P2</td>
<td>variableB</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S27_R1.5_counts.csv</td>
<td>variableA_N2_P1</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S28_R1.5_counts.csv</td>
<td>variableC_N1_ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S30_R1.5_counts.csv</td>
<td>variableB_N1_P2</td>
<td>variableB</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S31_R1.5_counts.csv</td>
<td>variableC_N2_P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S32_R1.5_counts.csv</td>
<td>variableA_N3_P1</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S33_R1.5_counts.csv</td>
<td>variableA_N3_ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S34_R1.5_counts.csv</td>
<td>variableC_N3_ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S35_R1.5_counts.csv</td>
<td>variableA_N1_P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S36_R1.5_counts.csv</td>
<td>variableA_N2_ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S37_R1.5_counts.csv</td>
<td>variableC_N2_ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S38_R1.5_counts.csv</td>
<td>variableA_N2_P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S39_R1.5_counts.csv</td>
<td>variableC_N1_P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S40_R1.5_counts.csv</td>
<td>variableB_N1_ctrl</td>
<td>variableB</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S41_R1.5_counts.csv</td>
<td>variableA_N1_P1</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S42_R1.5_counts.csv</td>
<td>variableA_N1_ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S43_R1.5_counts.csv</td>
<td>variableC_N3_P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S44_R1.5_counts.csv</td>
<td>variableB_N2_ctrl</td>
<td>variableB</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S45_R1.5_counts.csv</td>
<td>variableA_N3_P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
</tbody>
</table>
<p>-&gt;&gt;&gt; A simple way to format the metadata is to add three
columns, one for each experimental condition (i.e., termed “variable A
to C” in this example) in which the second experimental condition is
implemented (i.e., termed “ctrl, P1, P2” in this example). <br> Here is
an example of the first design of the novel metadata:</p>
<table style="width:95%;" class="table">
<colgroup>
<col width="19%">
<col width="19%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="11%">
<col width="9%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th>filename</th>
<th>samplename</th>
<th>variableA</th>
<th>variableB</th>
<th>variableC</th>
<th>condition</th>
<th>treatment</th>
<th>lib</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>S26_R1.5_counts.csv</td>
<td>variableB_N2_P2</td>
<td>NA</td>
<td>P2</td>
<td>NA</td>
<td>variableB</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S27_R1.5_counts.csv</td>
<td>variableA_N2_P1</td>
<td>P1</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S28_R1.5_counts.csv</td>
<td>variableC_N1_ctrl</td>
<td>NA</td>
<td>NA</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S30_R1.5_counts.csv</td>
<td>variableB_N1_P2</td>
<td>NA</td>
<td>P2</td>
<td>NA</td>
<td>variableB</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S31_R1.5_counts.csv</td>
<td>variableC_N2_P2</td>
<td>NA</td>
<td>NA</td>
<td>P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S32_R1.5_counts.csv</td>
<td>variableA_N3_P1</td>
<td>P1</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S33_R1.5_counts.csv</td>
<td>variableA_N3_ctrl</td>
<td>ctrl</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S34_R1.5_counts.csv</td>
<td>variableC_N3_ctrl</td>
<td>NA</td>
<td>NA</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S35_R1.5_counts.csv</td>
<td>variableA_N1_P2</td>
<td>P2</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S36_R1.5_counts.csv</td>
<td>variableA_N2_ctrl</td>
<td>ctrl</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S37_R1.5_counts.csv</td>
<td>variableC_N1_ctrl</td>
<td>NA</td>
<td>NA</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S38_R1.5_counts.csv</td>
<td>variableA_N2_P2</td>
<td>P2</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S39_R1.5_counts.csv</td>
<td>variableC_N2_P2</td>
<td>NA</td>
<td>NA</td>
<td>P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S40_R1.5_counts.csv</td>
<td>variableB_N1_ctrl</td>
<td>NA</td>
<td>ctrl</td>
<td>NA</td>
<td>variableB</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S41_R1.5_counts.csv</td>
<td>variableA_N1_P1</td>
<td>P1</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S42_R1.5_counts.csv</td>
<td>variableA_N1_ctrl</td>
<td>ctrl</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S43_R1.5_counts.csv</td>
<td>variableC_N3_P2</td>
<td>NA</td>
<td>NA</td>
<td>P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S44_R1.5_counts.csv</td>
<td>variableB_N3_ctrl</td>
<td>NA</td>
<td>ctrl</td>
<td>NA</td>
<td>variableB</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S45_R1.5_counts.csv</td>
<td>variableA_N3_P2</td>
<td>P2</td>
<td>NA</td>
<td>NA</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
</tbody>
</table>
<p>-&gt;&gt;&gt; If the analysis includes comparison of 2 conditions
independently of the other (e.g., all ctrl vs all P2), the addition of a
single column is required (i.e., “comp1” in this example).<br> Here is
an example of the second design of the novel metadata:</p>
<table style="width:96%;" class="table">
<colgroup>
<col width="26%">
<col width="25%">
<col width="8%">
<col width="13%">
<col width="13%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th>filename</th>
<th>samplename</th>
<th>comp1</th>
<th>condition</th>
<th>treatment</th>
<th>lib</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>S26_R1.5_counts.csv</td>
<td>variableB_N2_P2</td>
<td>P2</td>
<td>variableB</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S27_R1.5_counts.csv</td>
<td>variableA_N2_P1</td>
<td>NA</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S28_R1.5_counts.csv</td>
<td>variableC_N1_ctrl</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S30_R1.5_counts.csv</td>
<td>variableB_N1_P2</td>
<td>P2</td>
<td>variableB</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S31_R1.5_counts.csv</td>
<td>variableC_N2_P2</td>
<td>P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S32_R1.5_counts.csv</td>
<td>variableA_N3_P1</td>
<td>NA</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S33_R1.5_counts.csv</td>
<td>variableA_N3_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S34_R1.5_counts.csv</td>
<td>variableC_N3_ctrl</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S35_R1.5_counts.csv</td>
<td>variableA_N1_P2</td>
<td>P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S36_R1.5_counts.csv</td>
<td>variableA_N2_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S37_R1.5_counts.csv</td>
<td>variableC_N1_ctrl</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S38_R1.5_counts.csv</td>
<td>variableA_N2_P2</td>
<td>P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S39_R1.5_counts.csv</td>
<td>variableC_N2_P2</td>
<td>P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S40_R1.5_counts.csv</td>
<td>variableB_N1_ctrl</td>
<td>ctrl</td>
<td>variableB</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S41_R1.5_counts.csv</td>
<td>variableA_N1_P1</td>
<td>NA</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S42_R1.5_counts.csv</td>
<td>variableA_N1_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S43_R1.5_counts.csv</td>
<td>variableC_N3_P2</td>
<td>P2</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S44_R1.5_counts.csv</td>
<td>variableB_N3_ctrl</td>
<td>ctrl</td>
<td>variableB</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S45_R1.5_counts.csv</td>
<td>variableA_N3_P2</td>
<td>P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
</tbody>
</table>
<p>-&gt;&gt;&gt; If the analysis includes comparison of a pool of 2
conditions independently of the other (e.g., all ctrl vs all P1+P2), the
addition of a single column is required (i.e., “comp1” in this example)
with a new annotation (i.e., “P” = P1 or P2).<br> Here is an example of
the third design of the novel metadata:</p>
<table style="width:96%;" class="table">
<colgroup>
<col width="26%">
<col width="25%">
<col width="8%">
<col width="13%">
<col width="13%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th>filename</th>
<th>samplename</th>
<th>comp1</th>
<th>condition</th>
<th>treatment</th>
<th>lib</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>S26_R1.5_counts.csv</td>
<td>variableB_N2_P2</td>
<td>P</td>
<td>variableB</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S27_R1.5_counts.csv</td>
<td>variableA_N2_P1</td>
<td>P</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S28_R1.5_counts.csv</td>
<td>variableC_N1_ctrl</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S30_R1.5_counts.csv</td>
<td>variableB_N1_P2</td>
<td>P</td>
<td>variableB</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S31_R1.5_counts.csv</td>
<td>variableC_N2_P2</td>
<td>P</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S32_R1.5_counts.csv</td>
<td>variableA_N3_P1</td>
<td>P</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S33_R1.5_counts.csv</td>
<td>variableA_N3_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S34_R1.5_counts.csv</td>
<td>variableC_N3_ctrl</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S35_R1.5_counts.csv</td>
<td>variableA_N1_P2</td>
<td>P</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S36_R1.5_counts.csv</td>
<td>variableA_N2_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S37_R1.5_counts.csv</td>
<td>variableC_N1_ctrl</td>
<td>ctrl</td>
<td>variableC</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S38_R1.5_counts.csv</td>
<td>variableA_N2_P2</td>
<td>P</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S39_R1.5_counts.csv</td>
<td>variableC_N2_P2</td>
<td>P</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S40_R1.5_counts.csv</td>
<td>variableB_N1_ctrl</td>
<td>ctrl</td>
<td>variableB</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S41_R1.5_counts.csv</td>
<td>variableA_N1_P1</td>
<td>P</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S42_R1.5_counts.csv</td>
<td>variableA_N1_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S43_R1.5_counts.csv</td>
<td>variableC_N3_P2</td>
<td>P</td>
<td>variableC</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S44_R1.5_counts.csv</td>
<td>variableB_N3_ctrl</td>
<td>ctrl</td>
<td>variableB</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S45_R1.5_counts.csv</td>
<td>variableA_N3_P2</td>
<td>P</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 class="unnumbered" id="ii-create-a-new-riboclass">(ii) create a new Riboclass:<a class="anchor" aria-label="anchor" href="#ii-create-a-new-riboclass"></a>
</h4>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_exp_design</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_ribodata.html">load_ribodata</a></span><span class="op">(</span>count_path <span class="op">=</span> <span class="va">...</span>, metadata <span class="op">=</span> <span class="st">"./new_metadata.csv"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 class="unnumbered" id="comparison-table">Comparison table<a class="anchor" aria-label="anchor" href="#comparison-table"></a>
</h4>
<p>To perform an automated multiple group comparison using the
<em><code>report_diff_sites</code></em> function, a dataframe containing
all the biological comparisons should be prepared. This dataframe is
created using the code provided below and including in a new RiboClass
created using the <em><code><a href="../reference/load_ribodata.html">load_ribodata()</a></code></em> function and the
argument comparison.</p>
<p>Here is an example:</p>
<p>The study includes 3 experimental conditions termed “variable A to C”
(e.g., wild-type, mutation 1, mutation 2) that has been exposed to
another condition termed “ctrl, P1 and P2 (e.g., untreated vs treatment
with 2 different concentrations). All these experimental conditions have
been performed in replicates termed “N”. In summary, it includes:<br> -
VariableA has 1 control (ctrl) and 2 cases (P1 and P2) with 3 replicates
each (total = 9 samples)<br> - VariableB has 1 control (ctrl) and 1 case
(P2) with 2 replicates each (total = 4 samples)<br> - VariableC has 1
control (ctrl) and 1 case (P2) with 3 replicates or each (total = 6
samples)<br><br> The objectives are to perform 2 comparisons in the
“variableA” condition only, as summarized here:</p>
<table style="width:53%;" class="table">
<colgroup>
<col width="16%">
<col width="18%">
<col width="18%">
</colgroup>
<thead><tr class="header">
<th>
<p>comp</p>
<p>(mandatory)</p>
</th>
<th>
<p>ctrl</p>
<p>(mandatory)</p>
</th>
<th>
<p>cases</p>
<p>(mandatory)</p>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>comp1 (mandatory)</td>
<td>ctrl</td>
<td>P1</td>
</tr>
<tr class="even">
<td>comp2 (optional)</td>
<td>ctrl</td>
<td>P2</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 class="unnumbered" id="i-creation-of-the-dataframe-for-the-report_diff_sites-function">(i) Creation of the dataframe for the
report_diff_sites function:<a class="anchor" aria-label="anchor" href="#i-creation-of-the-dataframe-for-the-report_diff_sites-function"></a>
</h4>
<p>Here is the code to create the comparison table</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#---------------------Creation of the comparison table--------------------------</span></span>
<span><span class="co"># Here is the comparison table needed for this example data</span></span>
<span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>  comp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"comp1"</span>, <span class="st">"comp2"</span><span class="op">)</span>,</span>
<span>  ctrl <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ctrl"</span>, <span class="st">"ctrl"</span><span class="op">)</span>,</span>
<span>  cases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P1"</span>, <span class="st">"P2"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 class="unnumbered" id="ii-creation-of-a-new-riboclass-including-only-the-required-samples">(ii) Creation of a new RiboClass including only
the required samples<a class="anchor" aria-label="anchor" href="#ii-creation-of-a-new-riboclass-including-only-the-required-samples"></a>
</h4>
<p>Here is an example to extract a new RiboClass incuding only the
samples required to perfom the multiple comparison. It refers to the
metadata/RiboClass created in the previous section “(i) Update of the
metadata .csv file”, in the subsection “If the analysis includes
comparison of 2 conditions independently of the other”.</p>
<p>Store the subset samples required for the analysis using the column
names of the metadata created in the previous section “Addition of new
columns in the metadata”</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kept_samples</span> <span class="op">&lt;-</span> <span class="va">ribo_adj_annot</span><span class="op">$</span><span class="va">metadata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co">#use the new RiboClass created in the previous section “Addition of new columns in the metadata”  </span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">variableA</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># filter the lines in the metadata to keep only those that are not NA in the "variableA" column</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">samplename</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create a copy of the RiboClass with the subset of samples</span></span>
<span><span class="va">ribo_adj_annot_variableA</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/keep_ribo_samples.html">keep_ribo_samples</a></span><span class="op">(</span><span class="va">ribo_adj_annot</span>,<span class="va">kept_samples</span><span class="op">)</span> <span class="co"># save the new RiboClass including only the samples with information in the “variableA” column</span></span></code></pre></div>
<table style="width:96%;" class="table">
<colgroup>
<col width="25%">
<col width="24%">
<col width="12%">
<col width="14%">
<col width="12%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th>filename</th>
<th>samplename</th>
<th>variableA</th>
<th>treatment</th>
<th>condition</th>
<th>lib</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>S27_R1.5_counts.csv</td>
<td>variableA_N2_P1</td>
<td>P1</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S32_R1.5_counts.csv</td>
<td>variableA_N3_P1</td>
<td>P1</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S33_R1.5_counts.csv</td>
<td>variableA_N3_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S35_R1.5_counts.csv</td>
<td>variableA_N1_P2</td>
<td>P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S36_R1.5_counts.csv</td>
<td>variableA_N2_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S38_R1.5_counts.csv</td>
<td>variableA_N2_P2</td>
<td>P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S41_R1.5_counts.csv</td>
<td>variableA_N1_P1</td>
<td>P1</td>
<td>variableA</td>
<td>P1</td>
<td>L2</td>
</tr>
<tr class="even">
<td>S42_R1.5_counts.csv</td>
<td>variableA_N1_ctrl</td>
<td>ctrl</td>
<td>variableA</td>
<td>ctrl</td>
<td>L2</td>
</tr>
<tr class="odd">
<td>S45_R1.5_counts.csv</td>
<td>variableA_N3_P2</td>
<td>P2</td>
<td>variableA</td>
<td>P2</td>
<td>L2</td>
</tr>
</tbody>
</table>
<p>Note: the metadata should contain as many columns as there are
comparisons expected for the analysis.</p>
<p>Note: If there is only one column condition, it is still the same
thing. Here in the metadata example we can see 3 column variables. They
are the different comparisons that are wanted to be analysed. You can
replace the name for your custom ones. You can also run the analysis on
the “condition” column. In this case analysis will be done comparing as
many modalities as you have in the same report (!! be aware that even in
this case you need a comparison table for the diff_site analysis). (The
treatment column is optional)</p>
<p>The <a href="#ome-site-by-site-report">2’Ome site-by-site report</a>
can now be used.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="ome-profiling-report">2’Ome profiling report<a class="anchor" aria-label="anchor" href="#ome-profiling-report"></a>
</h2>
<p>A 2’Ome profiling report (2ome_analysis) can be automatically
generated. It uses several metrics to help identify global profile of
methylation between biological conditions.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/report_2ome_sites.html">report_2ome_sites</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo_adjusted</span>, specie <span class="op">=</span> <span class="st">"human"</span>, condition_col <span class="op">=</span> <span class="st">"condition"</span>, project_name <span class="op">=</span> <span class="st">"name_of_project"</span>, comments <span class="op">=</span> <span class="st">"./path/to/comments_for_this_report.Rmd"</span><span class="op">)</span></span></code></pre></div>
<p>To determine whether 2’Ome profiles are different between conditions
using either all the annotated 2’Ome sites or only the most variable
2’Ome sites, several functions have been implemented to obtain an html
report. Here is a list of the implemented plots/dataframe with the
function name:</p>
<ul>
<li>principal component analysis using the C-scores of all or the most
variable annotated 2’Ome sites (<em><code><a href="../reference/plot_pca.html">plot_pca()</a></code></em>)</li>
<li>heatmap using the C-scores of all or the most variable the annotated
2’Ome sites (<em><code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code></em>)</li>
<li>boxplot using the C-scores of the annotated 2’Ome sites in
descending order of variability
(<em><code><a href="../reference/boxplot_cscores.html">boxplot_cscores()</a></code></em>)</li>
<li>scatterplot using the Interquartile Range (IQR) of the C-scores of
all the annotated 2’Ome sites in descending order of variability with a
Gauss curve given the IQR distribution
(<em><code><a href="../reference/plot_sites_by_IQR.html">plot_sites_by_IQR()</a></code></em>)<br>
</li>
<li>dataframe providing the metrics used to generate the report
<em>(<code><a href="../reference/get_2ome_summary.html">get_2ome_summary()</a></code>)</em>
</li>
</ul>
<p>Note: Moreover, the argument <code>only_annotated</code>, which is
included in all the plot related functions, enables the plotting of only
the annotated rRNA 2’Ome sites of interest (i.e., with biological
relevance) when set to true.</p>
<div class="section level4">
<h4 class="unnumbered" id="pca">PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h4>
<p>As an example, the <em><code><a href="../reference/plot_pca.html">plot_pca()</a></code></em> function is
presented below.</p>
<p>Here is an example comparing samples reflecting different biological
conditions based on the rRNA 2’Ome profile of the provided
human_methylated list:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>         color_col <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>         only_annotated <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-27-1.png" width="700"></p>
<p>Note: by default, PC1 and PC2 axes are plotted. However, the user can
choose the PCA axes of interest using the “axes” argument.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>         color_col <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>         axes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>, <span class="co">#PC2 and PC3 will be plotted</span></span>
<span>         only_annotated <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-28-1.png" width="700"></p>
<p>Note: the function returns the complete dudi.pca <em>object</em>
(<code>dudi.pca()</code> function from ade4 library) instead of the plot
by setting <em>object_only</em> to TRUE:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>         color_col <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>         only_annotated <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         object_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 class="unnumbered" id="comments-1">Comments<a class="anchor" aria-label="anchor" href="#comments-1"></a>
</h4>
<p>A comment section is available in the 2’Ome profiling report to add
some personal comments. To do so, a Rmarkdown file should be created and
saved in the working directory. Then, path to this report should be
given in the comments argument of the
<em><code>report_2ome_site()</code></em> function.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/report_2ome_sites.html">report_2ome_sites</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>, specie <span class="op">=</span> <span class="st">"human"</span>, condition_col <span class="op">=</span> <span class="st">"condition"</span>, project_name <span class="op">=</span> <span class="st">"name_of_project"</span>, comments <span class="op">=</span>  <span class="st">"./path/to/comments_for_this_report.Rmd"</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="ome-site-by-site-report">2’Ome site-by-site report<a class="anchor" aria-label="anchor" href="#ome-site-by-site-report"></a>
</h2>
<p>A site-by-site 2’Ome comparison report can be automatically
generated. It uses several metrics and statistical tests to identify
differentially 2’Ome sites between biological conditions. To run these
functions, a <a href="#comparison-table">Comparison table</a> and an
updated metadata/RiboClass needs to be generated (see <a href="#analysis-design-for-multiple-comparisons">Analysis design for
multiple comparisons</a>)</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/report_diff_sites.html">report_diff_sites</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo_adjusted</span>, specie <span class="op">=</span> <span class="st">"human"</span>, condition_col <span class="op">=</span> <span class="st">"comp1"</span>, project_name <span class="op">=</span> <span class="st">"name"</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span>, comments <span class="op">=</span> <span class="st">"./path/to/comments_for_this_report.Rmd"</span><span class="op">)</span> </span></code></pre></div>
<p>To determine whether 2’Ome levels is different between conditions at
each site, several functions have been implemented to obtain an html
report. Here is a list of the implemented plots/dataframe with the
function name:</p>
<ul>
<li>line plot using the mean C-score of all annotated 2’Ome sites
(<em><code><a href="../reference/plot_global_profile.html">plot_global_profile()</a></code></em>)<br>
</li>
<li>histogram plot representing differences in the median C-score
(<em><code><a href="../reference/plot_comparison_median.html">plot_comparison_median()</a></code></em>)<br>
</li>
<li>boxplot showing the mean C-scores for each conditions, but only for
sites where a statistical difference was observed (ANOVA, Welch,
Kruskal-Wallis, Wilcoxon) (<em><code><a href="../reference/plot_stat.html">plot_stat()</a></code></em>) <br>
</li>
<li>dataframe providing the metrics used to generate the report
<em>(<code><a href="../reference/get_diff_sites_summary.html">get_diff_sites_summary()</a></code>)</em>
</li>
</ul>
<div class="section level3">
<h3 class="unnumbered" id="summary-table-of-statistical-tests">Summary table of statistical tests<a class="anchor" aria-label="anchor" href="#summary-table-of-statistical-tests"></a>
</h3>
<p>To get a table of significant sites in at least one test the function
below can be used.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_diff_sites_summary.html">get_diff_sites_summary</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo_toy</span>, condition_col <span class="op">=</span> <span class="st">"comp1"</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 class="unnumbered" id="comments-2">Comments<a class="anchor" aria-label="anchor" href="#comments-2"></a>
</h3>
<p>A comment section is available in the site-by-site 2’Ome comparison
report to add some personal comments. To do so, a Rmarkdown file should
be created and saved in the working directory. Then, path to this report
should be given in the <code>comments</code> argument of the
<em><code><a href="../reference/report_diff_sites.html">report_diff_sites()</a></code></em> function.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/report_diff_sites.html">report_diff_sites</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo_toy</span>, specie <span class="op">=</span> <span class="st">"human"</span>, condition_col <span class="op">=</span> <span class="st">"comp1"</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span>, comments <span class="op">=</span> <span class="st">"./path/to/comments_for_this_report.Rmd"</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="exporting-data">Exporting data<a class="anchor" aria-label="anchor" href="#exporting-data"></a>
</h2>
<p>Data can be exported as two different objects in three different
ways.</p>
<div class="section level3">
<h3 id="export-as-a-dataframe">Export as a dataframe<a class="anchor" aria-label="anchor" href="#export-as-a-dataframe"></a>
</h3>
<p>The user can export data as a dataframe using the
<em><code><a href="../reference/extract_data.html">extract_data()</a></code></em> function.</p>
<p>By default, C-scores for all the genomic rRNA positions are
exported.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_data.html">extract_data</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                        col <span class="op">=</span> <span class="st">"cscore"</span><span class="op">)</span></span></code></pre></div>
<p>The user can export data related to the subset of annotated rRNA
2’Ome sites by setting the <code>only_annotated</code> argument to
TRUE.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_data.html">extract_data</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                        col <span class="op">=</span> <span class="st">"cscore"</span>,</span>
<span>                        only_annotated <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Excerpt from the output dataframe, where S1, S2 and S3 are
samples:</caption>
<thead><tr class="header">
<th align="left">site</th>
<th align="right">S1</th>
<th align="right">S2</th>
<th align="right">S3</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">18S_Am27</td>
<td align="right">0.9783333</td>
<td align="right">0.9745514</td>
<td align="right">0.9810412</td>
</tr>
<tr class="even">
<td align="left">18S_Am99</td>
<td align="right">0.9680204</td>
<td align="right">0.9646470</td>
<td align="right">0.9724150</td>
</tr>
<tr class="odd">
<td align="left">18S_Um116</td>
<td align="right">0.9276274</td>
<td align="right">0.9470968</td>
<td align="right">0.9407083</td>
</tr>
<tr class="even">
<td align="left">18S_Um121</td>
<td align="right">0.9630216</td>
<td align="right">0.9684459</td>
<td align="right">0.9691576</td>
</tr>
<tr class="odd">
<td align="left">18S_Am159</td>
<td align="right">0.9629730</td>
<td align="right">0.9620986</td>
<td align="right">0.9686766</td>
</tr>
<tr class="even">
<td align="left">18S_Am166</td>
<td align="right">0.9809492</td>
<td align="right">0.9744627</td>
<td align="right">0.9744998</td>
</tr>
<tr class="odd">
<td align="left">18S_Um172</td>
<td align="right">0.9510189</td>
<td align="right">0.9400922</td>
<td align="right">0.9517981</td>
</tr>
<tr class="even">
<td align="left">18S_Cm174</td>
<td align="right">0.9119119</td>
<td align="right">0.8668456</td>
<td align="right">0.8806886</td>
</tr>
<tr class="odd">
<td align="left">18S_Um354</td>
<td align="right">0.9709273</td>
<td align="right">0.9722054</td>
<td align="right">0.9756174</td>
</tr>
<tr class="even">
<td align="left">18S_Um428</td>
<td align="right">0.9268668</td>
<td align="right">0.9276018</td>
<td align="right">0.9504224</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 class="unnumbered" id="export-as-a-ggplot-ready-dataframe">Export as a ggplot-ready dataframe<a class="anchor" aria-label="anchor" href="#export-as-a-ggplot-ready-dataframe"></a>
</h3>
<p>The user can export data as a ggplot-ready dataframe using the
<em><code><a href="../reference/format_to_plot.html">format_to_plot()</a></code></em> function.</p>
<p>By default, C-scores for all the genomic rRNA positions are exported.
The user can export additional information contained in the metadata by
specifying the name of the column of interest. The user can export
information related to the subset of annotated rRNA 2’Ome sites by
setting the <em>only_annotated</em> argument to TRUE.</p>
<p>Here is an example of ggplot-ready dataframe containing the C-scores
of all the genomic rRNA positions as well as the condition related to
the particular sample of interest:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ggplot_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_to_plot.html">format_to_plot</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo_adjusted</span>, metadata_col <span class="op">=</span> <span class="st">"condition"</span>, only_annotated <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Excerpt from the output ggplot-ready dataframe</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="left">site</th>
<th align="right">cscore</th>
<th align="left">condition</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">501</td>
<td align="left">RNA1</td>
<td align="left">18S_0501</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">502</td>
<td align="left">RNA1</td>
<td align="left">18S_0502</td>
<td align="right">0.2601804</td>
<td align="left">RNA ref</td>
</tr>
<tr class="odd">
<td align="left">503</td>
<td align="left">RNA1</td>
<td align="left">18S_0503</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">504</td>
<td align="left">RNA1</td>
<td align="left">18S_0504</td>
<td align="right">0.3179712</td>
<td align="left">RNA ref</td>
</tr>
<tr class="odd">
<td align="left">505</td>
<td align="left">RNA1</td>
<td align="left">18S_0505</td>
<td align="right">0.5132894</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">506</td>
<td align="left">RNA1</td>
<td align="left">18S_0506</td>
<td align="right">0.6354548</td>
<td align="left">RNA ref</td>
</tr>
<tr class="odd">
<td align="left">507</td>
<td align="left">RNA1</td>
<td align="left">18S_0507</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">508</td>
<td align="left">RNA1</td>
<td align="left">18S_0508</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="odd">
<td align="left">509</td>
<td align="left">RNA1</td>
<td align="left">18S_0509</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">510</td>
<td align="left">RNA1</td>
<td align="left">18S_0510</td>
<td align="right">0.9380812</td>
<td align="left">RNA ref</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 class="unnumbered" id="export-as-a-dataframe-by-condition">Export as a dataframe by condition<a class="anchor" aria-label="anchor" href="#export-as-a-dataframe-by-condition"></a>
</h3>
<p>The <em><code><a href="../reference/mean_samples_by_condition.html">mean_samples_by_condition()</a></code></em> function
regroup all the genomic rRNA positions by condition and compute mean and
standard deviation for each condition. Then it displays a table.</p>
<p>By default, the function provides the metrics of the C-scores. By
setting the <em>value</em> argument to “count”, it is also possible to
calculate these values from the read-end counts. Calculations can be
limited to only annotated rRNA 2’Ome sites by setting the
<em>only_annotated</em> argument to TRUE.</p>
<p>Here is an example of dataframe showing the summarized C-scores per
modality of the “condition” variable for all the genomic rRNA
positions:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mean_tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mean_samples_by_condition.html">mean_samples_by_condition</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                                    value <span class="op">=</span> <span class="st">"cscore"</span>, <span class="co">#or "count"</span></span>
<span>                                    metadata_condition <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                                    only_annotated <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Excerpt from the output dataframe by condition</caption>
<thead><tr class="header">
<th align="left">site</th>
<th align="left">condition</th>
<th align="right">mean</th>
<th align="right">sd</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">18S_Am1031</td>
<td align="left">RNA ref</td>
<td align="right">0.9749922</td>
<td align="right">0.0047461</td>
</tr>
<tr class="even">
<td align="left">18S_Am1031</td>
<td align="left">cond1</td>
<td align="right">0.9703699</td>
<td align="right">0.0032725</td>
</tr>
<tr class="odd">
<td align="left">18S_Am1031</td>
<td align="left">cond2</td>
<td align="right">0.9723714</td>
<td align="right">0.0057723</td>
</tr>
<tr class="even">
<td align="left">18S_Am1383</td>
<td align="left">RNA ref</td>
<td align="right">0.9796876</td>
<td align="right">0.0035939</td>
</tr>
<tr class="odd">
<td align="left">18S_Am1383</td>
<td align="left">cond1</td>
<td align="right">0.9765212</td>
<td align="right">0.0027083</td>
</tr>
<tr class="even">
<td align="left">18S_Am1383</td>
<td align="left">cond2</td>
<td align="right">0.9779684</td>
<td align="right">0.0028759</td>
</tr>
<tr class="odd">
<td align="left">18S_Am159</td>
<td align="left">RNA ref</td>
<td align="right">0.9684126</td>
<td align="right">0.0022025</td>
</tr>
<tr class="even">
<td align="left">18S_Am159</td>
<td align="left">cond1</td>
<td align="right">0.9609920</td>
<td align="right">0.0054882</td>
</tr>
<tr class="odd">
<td align="left">18S_Am159</td>
<td align="left">cond2</td>
<td align="right">0.9623362</td>
<td align="right">0.0065239</td>
</tr>
<tr class="even">
<td align="left">18S_Am166</td>
<td align="left">RNA ref</td>
<td align="right">0.9826490</td>
<td align="right">0.0033169</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] rRMSAnalyzer_3.0.0 kableExtra_1.4.0   DT_0.34.0          dplyr_1.1.4       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] colorRamp2_0.1.0        DBI_1.2.3               rlang_1.1.6            </span></span>
<span><span class="co">#&gt;   [4] magrittr_2.0.3          clue_0.3-66             GetoptLong_1.0.5       </span></span>
<span><span class="co">#&gt;   [7] ade4_1.7-23             matrixStats_1.5.0       compiler_4.5.1         </span></span>
<span><span class="co">#&gt;  [10] RSQLite_2.4.3           mgcv_1.9-3              png_0.1-8              </span></span>
<span><span class="co">#&gt;  [13] systemfonts_1.2.3       vctrs_0.6.5             sva_3.56.0             </span></span>
<span><span class="co">#&gt;  [16] stringr_1.5.1           pkgconfig_2.0.3         shape_1.4.6.1          </span></span>
<span><span class="co">#&gt;  [19] crayon_1.5.3            fastmap_1.2.0           XVector_0.48.0         </span></span>
<span><span class="co">#&gt;  [22] backports_1.5.0         labeling_0.4.3          rmarkdown_2.29         </span></span>
<span><span class="co">#&gt;  [25] UCSC.utils_1.4.0        ragg_1.5.0              purrr_1.1.0            </span></span>
<span><span class="co">#&gt;  [28] bit_4.6.0               xfun_0.53               cachem_1.1.0           </span></span>
<span><span class="co">#&gt;  [31] GenomeInfoDb_1.44.2     jsonlite_2.0.0          blob_1.2.4             </span></span>
<span><span class="co">#&gt;  [34] BiocParallel_1.42.1     broom_1.0.9             parallel_4.5.1         </span></span>
<span><span class="co">#&gt;  [37] cluster_2.1.8.1         R6_2.6.1                bslib_0.9.0            </span></span>
<span><span class="co">#&gt;  [40] stringi_1.8.7           RColorBrewer_1.1-3      limma_3.64.3           </span></span>
<span><span class="co">#&gt;  [43] genefilter_1.90.0       car_3.1-3               jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt;  [46] Rcpp_1.1.0              iterators_1.0.14        knitr_1.50             </span></span>
<span><span class="co">#&gt;  [49] IRanges_2.42.0          Matrix_1.7-3            splines_4.5.1          </span></span>
<span><span class="co">#&gt;  [52] tidyselect_1.2.1        rstudioapi_0.17.1       abind_1.4-8            </span></span>
<span><span class="co">#&gt;  [55] yaml_2.3.10             doParallel_1.0.17       codetools_0.2-20       </span></span>
<span><span class="co">#&gt;  [58] lattice_0.22-7          tibble_3.3.0            KEGGREST_1.48.1        </span></span>
<span><span class="co">#&gt;  [61] Biobase_2.68.0          withr_3.0.2             evaluate_1.0.5         </span></span>
<span><span class="co">#&gt;  [64] survival_3.8-3          desc_1.4.3              xml2_1.4.0             </span></span>
<span><span class="co">#&gt;  [67] Biostrings_2.76.0       circlize_0.4.16         pillar_1.11.0          </span></span>
<span><span class="co">#&gt;  [70] ggpubr_0.6.1            MatrixGenerics_1.20.0   carData_3.0-5          </span></span>
<span><span class="co">#&gt;  [73] foreach_1.5.2           stats4_4.5.1            generics_0.1.4         </span></span>
<span><span class="co">#&gt;  [76] S4Vectors_0.46.0        ggplot2_3.5.2           scales_1.4.0           </span></span>
<span><span class="co">#&gt;  [79] xtable_1.8-4            glue_1.8.0              tools_4.5.1            </span></span>
<span><span class="co">#&gt;  [82] locfit_1.5-9.12         annotate_1.86.1         ggsignif_0.6.4         </span></span>
<span><span class="co">#&gt;  [85] fs_1.6.6                XML_3.99-0.19           grid_4.5.1             </span></span>
<span><span class="co">#&gt;  [88] tidyr_1.3.1             edgeR_4.6.3             AnnotationDbi_1.70.0   </span></span>
<span><span class="co">#&gt;  [91] colorspace_2.1-1        nlme_3.1-168            GenomeInfoDbData_1.2.14</span></span>
<span><span class="co">#&gt;  [94] Formula_1.2-5           cli_3.6.5               textshaping_1.0.3      </span></span>
<span><span class="co">#&gt;  [97] viridisLite_0.4.2       svglite_2.2.1           ComplexHeatmap_2.24.1  </span></span>
<span><span class="co">#&gt; [100] gtable_0.3.6            rstatix_0.7.2           sass_0.4.10            </span></span>
<span><span class="co">#&gt; [103] digest_0.6.37           BiocGenerics_0.54.0     ggrepel_0.9.6          </span></span>
<span><span class="co">#&gt; [106] rjson_0.2.23            htmlwidgets_1.6.4       farver_2.1.2           </span></span>
<span><span class="co">#&gt; [109] memoise_2.0.1           htmltools_0.5.8.1       pkgdown_2.1.3          </span></span>
<span><span class="co">#&gt; [112] factoextra_1.0.7        lifecycle_1.0.4         httr_1.4.7             </span></span>
<span><span class="co">#&gt; [115] statmod_1.5.0           GlobalOptions_0.1.2     bit64_4.6.0-1          </span></span>
<span><span class="co">#&gt; [118] MASS_7.3-65</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-birkedal2014" class="csl-entry">
Birkedal, Ulf, Mikkel Christensen-Dalsgaard, Nicolai Krogh,
Radhakrishnan Sabarinathan, Jan Gorodkin, and Henrik Nielsen. 2014.
<span>“Profiling of Ribose Methylations in RNA by High-Throughput
Sequencing.”</span> <em>Angewandte Chemie International Edition</em>,
November, n/a–. <a href="https://doi.org/10.1002/anie.201408362" class="external-link">https://doi.org/10.1002/anie.201408362</a>.
</div>
<div id="ref-Erales2017" class="csl-entry">
Erales, Jenny, Virginie Marchand, Baptiste Panthu, Sandra Gillot,
Stéphane Belin, Sandra E. Ghayad, Maxime Garcia, et al. 2017.
<span>“Evidence for rRNA 2&amp;#x2032;-o-Methylation Plasticity: Control
of Intrinsic Translational Capabilities of Human Ribosomes.”</span>
<em>Proceedings of the National Academy of Sciences</em> 114 (49):
12934–39. <a href="https://doi.org/10.1073/pnas.1707674114" class="external-link">https://doi.org/10.1073/pnas.1707674114</a>.
</div>
<div id="ref-Marcel2021" class="csl-entry">
Marcel, Virginie, Janice Kielbassa, Virginie Marchand, Kundhavai S
Natchiar, Hermes Paraqindes, Flora Nguyen Van Long, Lilia Ayadi, et al.
2021. <span>“<span class="nocase">Ribosomal RNA 2’ O-methylation as a
novel layer of inter-tumour heterogeneity in breast
cancer</span>.”</span> <em>NAR Cancer</em> 3 (1). <a href="https://doi.org/10.1093/narcan/zcab006" class="external-link">https://doi.org/10.1093/narcan/zcab006</a>.
</div>
<div id="ref-Marchand2016" class="csl-entry">
Marchand, Virginie, Florence Blanloeil-Oillo, Mark Helm, and Yuri
Motorin. 2016. <span>“<span class="nocase">Illumina-based RiboMethSeq
approach for mapping of 2’-O-Me residues in RNA</span>.”</span>
<em>Nucleic Acids Research</em> 44 (16): e135–35. <a href="https://doi.org/10.1093/nar/gkw547" class="external-link">https://doi.org/10.1093/nar/gkw547</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Théo Combe, Hermes Paraqindes, Allyson Moureaux, Janice Kielbassa, Fleur Bourdelais, Sébastien Durand, Émilie Thomas, Anthony Ferrari, Virginie Marcel, Fondation Synergie Lyon Cancer, INCa, SIRIC program, LabEX program, Ligue Nationale Contre le Cancer.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
