<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="rRMSAnalyzer">
<title>rRMSAnalyzer: package to analyze RiboMethSeq data • rRMSAnalyzer</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="rRMSAnalyzer: package to analyze RiboMethSeq data">
<meta property="og:description" content="rRMSAnalyzer">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">rRMSAnalyzer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/rRMSAnalyzer.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/custom_anno.html">Create and use custom sites annotation</a>
    <a class="dropdown-item" href="../articles/custom_ribo.html">Customizing a RiboClass</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/RibosomeCRCL/rRMSAnalyzer/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>rRMSAnalyzer: package to analyze RiboMethSeq data</h1>
                        <h4 data-toc-skip class="author">Théo COMBE,
Hermes PARAQINDES, Janisse KIELBASSA, Emilie THOMAS, Anthony FERRARI and
Virginie MARCEL</h4>
            
            <h4 data-toc-skip class="date">06/23/2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RibosomeCRCL/rRMSAnalyzer/blob/HEAD/vignettes/rRMSAnalyzer.Rmd" class="external-link"><code>vignettes/rRMSAnalyzer.Rmd</code></a></small>
      <div class="d-none name"><code>rRMSAnalyzer.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>RiboMethSeq is an innovative RNAseq-based approach, which was
      developed in 2015 to analyze 2’O-ribose methylation (2’Ome) at all
      sites of ribosomal RNAs (rRNA) at once, in yeast <span class="citation">(Birkedal et al. 2014)</span>. This approach was
      then transferred to human using the Illumina technology <span class="citation">(Marchand et al. 2016; Erales et al. 2017; Marcel
      et al. 2021)</span>. Briefly, the presence of 2’Ome protects the
      phosphodiester bond located at the 3’ of the 2’Ome nucleotide from
      alkaline hydrolysis. Thus, the presence of 2’Ome at the given
      nucleotide n induces under-representation of RNA fragments
      starting at the nucleotide n+1 and an over-representation of RNA
      fragments ending at the nucleotide n-1, allowing the extrapolation
      of 2’Ome levels at the corresponding nucleotide position n (or
      C-score) varying from 0 to 1 <span class="citation">(Birkedal et
      al. 2014)</span>. <br><br> The rRMSAnalyzer package can be used
      for any kind of RNA and with all organisms. This package provides
      a set of user-friendly functions to compute C-scores from
      RiboMethSeq read end counts as input, adjust batch effect with
      ComBat-Seq, visualize the data and provide a table with the
      annotated human rRNA sites and their C-scores. Processing of the
      raw data to obtain read end counts from sequencing data was
      previously described <span class="citation">(Marchand et al.
      2016)</span>.</p>
    </div>
    
<p>The rRMSAnalyzer package can be used for any type of RNA and on all
organism. This package provides a set of user-friendly functions to
calculate C-scores from RiboMethSeq read end counts as input, perform
quality control of the dataset, adjust for potential batch effects, and
provide tools to visualize and analyze the data. In addition, a list of
potentially methylated human rRNA sites is provided for analyses.</p>
<p>Raw data processing to obtain read-end counts from sequencing data
can be performed using our Nextflow pipeline <a href="https://github.com/RibosomeCRCL/ribomethseq-nf" class="external-link">ribomethseq-nf</a>,
as already described <span class="citation">(Marchand et al.
2016)</span>.</p>
<div class="section level2">
<h2 id="help-bug-reports-and-suggestions">Help, bug reports and suggestions<a class="anchor" aria-label="anchor" href="#help-bug-reports-and-suggestions"></a>
</h2>
<p>To report a bug or any suggestion to improve the package, please let
us known by opening a new issue on: <em>github issue link coming
soon!</em></p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>We would like to thank all our collaborators for their advices and
suggestions.</p>
</div>
<div class="section level2">
<h2 id="funding">Funding<a class="anchor" aria-label="anchor" href="#funding"></a>
</h2>
<p>This project has been funded by the French Cancer Institute (INCa,
PLBIO 2019-138 MARACAS), the SIRIC Program (INCa-DGOS-Inserm_12563
LyRICAN) and Synergie Lyon Cancer Foundation.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The latest version of rRMSAnalyzer package can be installed from
Github with :</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"RibosomeCRCL/rRMSAnalyzer"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RibosomeCRCL/rRMSAnalyzer" class="external-link">rRMSAnalyzer</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_ribodata.html">load_ribodata</a></span><span class="op">(</span></span>
<span>              count_path <span class="op">=</span> <span class="st">"/path/to/your/csvfiles/directory/"</span>,</span>
<span>              metadata <span class="op">=</span> <span class="st">"path/to/metadata.csv"</span>,</span>
<span>              metadata_key <span class="op">=</span> <span class="st">"filename"</span>,</span>
<span>              metadata_id <span class="op">=</span> <span class="st">"samplename"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compute the c-score using different parameters,</span></span>
<span><span class="co"># including calculation of the local coverage using the mean instead of the median</span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_cscore.html">compute_cscore</a></span><span class="op">(</span><span class="va">ribo</span>, method <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If necessary, adjust any technical biases using ComBat-Seq.</span></span>
<span><span class="co"># Here, as an example, we use the "library" column in metadata.</span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/adjust_bias.html">adjust_bias</a></span><span class="op">(</span><span class="va">ribo</span>,<span class="st">"library"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot a Principal Component Analysis (PCA) whose colors depend on the "condition" column in metadata</span></span>
<span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo</span>,<span class="st">"condition"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="riboclass">RiboClass<a class="anchor" aria-label="anchor" href="#riboclass"></a>
</h2>
<p>RiboClass is the main class of the package that enables the storage
of both the data matrices (counts and c-scores) and the associated
metadata. It is automatically created when calling
<em>load_ribodata</em> (see <a href="#loading-data">Loading
data</a>).</p>
<p>It is a list containing three main elements, individually described
below:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Data</strong>: a list of dataframe, containing for each
sample the 5’ and/or 3’ read-end counts provided by the user, and the
calculated C-score.</p></li>
<li><p><strong>Metadata</strong>: a dataframe, containing all the
information related to the samples that can be provided by the
user.</p></li>
<li><p><strong>RNA_names</strong>: a dataframe, reporting the names of
the RNA used in Data.</p></li>
</ol>
<p>Some major function parameters (such as the normalization method used
for c-score computation) are also kept in the RiboClass object as a
reminder.</p>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<div class="section level3">
<h3 id="data-to-provide">Data to provide<a class="anchor" aria-label="anchor" href="#data-to-provide"></a>
</h3>
<div class="section level4">
<h4 id="read-end-counts">Read-end counts<a class="anchor" aria-label="anchor" href="#read-end-counts"></a>
</h4>
<p>To use this package, the user must provide at least one csv/tsv file
with the 5’, 3’ or 5’/3’ read end counts resulting from RiboMethSeq data
per sample. The folder structure containing the csv files is not
important, as long as either the directory and its sub-directories
contain the necessary csv/tsv files.</p>
<ol style="list-style-type: decimal">
<li><p>The <strong>name of the RNA</strong> on which the read end
counting was performed.</p></li>
<li><p>The <strong>number of the position</strong> on the RNA.</p></li>
<li><p>The <strong>value of the read end counts</strong> at the
position.</p></li>
</ol>
<p>Here is an example :</p>
<table class="table">
<thead><tr class="header">
<th>RNA</th>
<th>Position on RNA</th>
<th>read end count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>18S</td>
<td>123</td>
<td>3746</td>
</tr>
<tr class="even">
<td>18S</td>
<td>124</td>
<td>345</td>
</tr>
<tr class="odd">
<td>18S</td>
<td>125</td>
<td>324</td>
</tr>
<tr class="even">
<td>18S</td>
<td>126</td>
<td>789</td>
</tr>
<tr class="odd">
<td>18S</td>
<td>127</td>
<td>1234</td>
</tr>
</tbody>
</table>
<p>Note 1: it is not necessary to provide a header in the count files,
because column index can be used in the function <em>load_ribodata</em>,
using <em>count_value</em>, <em>count_rnaid</em> and <em>count_pos</em>
parameters.</p>
<p>Note 2: if no metadata is specified, rRMSAnalyzer will try to fetch
any csv files in the folder specified in count_path and its
subfolders.</p>
</div>
<div class="section level4">
<h4 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h4>
<p>The expected metadata is either a dataframe already in the R
environment or a csv/tsv file.</p>
<p><u>Two columns are mandatory for the metadata :</u></p>
<ol style="list-style-type: decimal">
<li><p><strong>filename</strong>: name of the csv file on disk
containing the read end counts described above. Do not modify it unless
the filename has changed on disk.</p></li>
<li><p><strong>samplename</strong>: rename the samples that will be
analyzed and displayed on the plots. This column can be modified, as
long as the sample names are unique.</p></li>
</ol>
<p>After these two mandatory columns, the user can provide as many
columns as needed for the analysis.</p>
<p>Here is an example of metadata for three samples:</p>
<table style="width:74%;" class="table">
<colgroup>
<col width="20%">
<col width="20%">
<col width="31%">
</colgroup>
<thead><tr class="header">
<th>
<p>filename</p>
<p>(mandatory)</p>
</th>
<th>
<p>samplename</p>
<p>(mandatory)</p>
</th>
<th>
<p>biological condition</p>
<p>(optionnal)</p>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample1.csv</td>
<td>sample 1</td>
<td>condition 1</td>
</tr>
<tr class="even">
<td>sample2.csv</td>
<td>sample 2</td>
<td>condition 1</td>
</tr>
<tr class="odd">
<td>sample3.csv</td>
<td>sample 3</td>
<td>condition 2</td>
</tr>
</tbody>
</table>
<p>Note: if no metadata is provided in load_ribodata, an empty metadata
will be created with the “filename” and “samplename” columns pre-filled.
The “samplename” column will be identical to “filename”, but can be
modified by the user.</p>
<p>Here is an example of auto-generated metadata:</p>
<table class="table">
<thead><tr class="header">
<th>filename</th>
<th>samplename</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample1.csv</td>
<td>sample1.csv</td>
</tr>
<tr class="even">
<td>sample2.csv</td>
<td>sample2.csv</td>
</tr>
<tr class="odd">
<td>sample3.csv</td>
<td>sample3.csv</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="how-to-load-the-data">How to load the data<a class="anchor" aria-label="anchor" href="#how-to-load-the-data"></a>
</h4>
<p>To load both data and metadata, and store them in a RiboClass, the
function <code>load_ribodata</code> is used.</p>
<p>The following example displays a call to load_ribodata, with all
parameters shown :</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"rRMSAnalyzer"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_ribodata.html">load_ribodata</a></span><span class="op">(</span></span>
<span>                      <span class="co">#data &amp; metadata files path</span></span>
<span>                      count_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">path</span>,<span class="st">"miniglioma/"</span><span class="op">)</span>,</span>
<span>                      metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">path</span>,<span class="st">"metadata.csv"</span><span class="op">)</span>,</span>
<span>                      <span class="co"># data &amp; metadata files separator</span></span>
<span>                      count_sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>                      metadata_sep <span class="op">=</span> <span class="st">","</span>,</span>
<span>                      <span class="co"># count data parameters :</span></span>
<span>                      count_header <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                      count_value <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                      count_rnaid <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                      count_pos <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                      <span class="co"># Metadata parameters :</span></span>
<span>                      metadata_key <span class="op">=</span> <span class="st">"filename"</span>,</span>
<span>                      metadata_id <span class="op">=</span> <span class="st">"samplename"</span>,</span>
<span>                      <span class="co"># c-score parameters :</span></span>
<span>                      flanking <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>                      ncores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Your data has been successfully loaded!</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="rna-names">RNA names<a class="anchor" aria-label="anchor" href="#rna-names"></a>
</h3>
<div class="section level4">
<h4 id="provided">Provided<a class="anchor" aria-label="anchor" href="#provided"></a>
</h4>
<p>RNA names are automatically obtained from the data and stored in a
generated dataframe inside the RiboClass. It contains two columns :</p>
<ol style="list-style-type: decimal">
<li><p><strong>original_name</strong>: original name of each RNA (e.g
NR_023363.1).</p></li>
<li><p><strong>current_name</strong>: current name of each RNA,
reflecting any user’s change with <em>rename_rna</em> function (see <a href="#rename-rna">Rename RNA</a>).</p></li>
</ol>
<p>This dataframe is used to keep track of the original name, which
often includes the NCBI’s accession ID.</p>
<p>Here is an example:</p>
<table class="table">
<thead><tr class="header">
<th>original_name</th>
<th>current_name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>NR_023363.1_5S</td>
<td>5S</td>
</tr>
<tr class="even">
<td>NR_046235.3_5.8S</td>
<td>5.8S</td>
</tr>
<tr class="odd">
<td>NR_046235.3_18S</td>
<td>18S</td>
</tr>
<tr class="even">
<td>NR_046235.3_28S</td>
<td>28S</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="custom">Custom<a class="anchor" aria-label="anchor" href="#custom"></a>
</h4>
<p>The user must not modify this dataframe manually. To rename or remove
RNA, the user can use rename_rna and remove_rna. The dataframe will be
updated accordingly.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="c-score-calculation">C-score calculation<a class="anchor" aria-label="anchor" href="#c-score-calculation"></a>
</h2>
<div class="section level3">
<h3 id="what-is-a-c-score">What is a C-score ?<a class="anchor" aria-label="anchor" href="#what-is-a-c-score"></a>
</h3>
<p>The C-score is a metric, calculated from the RiboMethSeq sequencing
data, used to evaluate the level of 2’O-ribose methylation (2’Ome) at a
given position in the RNA.</p>
<p>The C-score is calculated by normalizing the end read count with
respect to the local environment at each genomic position and directly
indicates the RNA 2’Ome level. The C-score ranges from 0 (i.e., no RNA
molecules of the sample are 2’Ome at this specific site) to 1 (i.e., all
RNA molecules of the sample are 2’Ome at this specific site). A C-score
with an intermediate value between 0 to 1 means that the sample contains
a mixture of un-2’Ome and 2’Ome RNA molecules.</p>
<p>To obtain a robust estimate of the 2’Ome level, different C-scores
can be determined depending on the parameters used to compute the local
coverage. In particular, the estimation method and the size of the local
coverage to be considered can be changed.</p>
<p>By default, the local coverage is estimated by calculating the
<strong>median</strong> of the end read counts in a <strong>flanking
region of 6</strong> (i.e., 6 nucleotides downstream the nucleotide n
and 6 upstream the nucleotide n, where n is the nucleotid directly
following the 2’Ome site of interest). This package provide the ability
to change these two parameters either when loading the data or during
the analysis.</p>
</div>
<div class="section level3">
<h3 id="c-score-computation-when-loading-data">C-score computation when loading data<a class="anchor" aria-label="anchor" href="#c-score-computation-when-loading-data"></a>
</h3>
<p>When using the load_ribodata function, a C-score is automatically
calculated for all genomic positions of the RNA. The C-score is computed
using either the default parameters of the load_ribodata function or
user-defined parameters as follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/load_ribodata.html">load_ribodata</a></span><span class="op">(</span>count.path <span class="op">=</span> <span class="st">"/path/to/csv/"</span>,</span>
<span>              metadata <span class="op">=</span> <span class="st">"/path/to/metadata.csv"</span>,</span>
<span>              <span class="co"># everything below is linked to c-score computation</span></span>
<span>              flanking <span class="op">=</span> <span class="fl">6</span>, <span class="co"># flanking region size</span></span>
<span>              method <span class="op">=</span> <span class="st">"median"</span>, <span class="co"># use mean or median on flanking region's values</span></span>
<span>              ncores <span class="op">=</span> <span class="fl">8</span> <span class="co"># number of CPU cores to use for computation</span></span>
<span>              <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="c-score-computation-during-the-analysis">C-score computation during the analysis<a class="anchor" aria-label="anchor" href="#c-score-computation-during-the-analysis"></a>
</h3>
<p>During the analysis, the C-score calculating parameters (method and
size of the flanking region) can be modified using the
<em>compute_cscore</em> function, which will automatically update the
C-score in the RiboClass.</p>
<p>In the following example, both the flanking region’s size of the
local coverage and the computation method have been modified:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_cscore.html">compute_cscore</a></span><span class="op">(</span><span class="va">ribo</span>,</span>
<span>                       flanking <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                       method <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span></span></code></pre></div>
<p><strong>Important: this function will override the previous c-score
of the RiboClass.</strong></p>
</div>
</div>
<div class="section level2">
<h2 id="quality-control">Quality control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<p>Due to technical limitations, it is sometimes necessary to conduct
wet-lab preparations of large cohorts in several batches. The main risk
when making several batches is to introduce <strong>technical
biases</strong> or <strong>batch effect</strong> in the dataset.</p>
<div class="section level3">
<h3 id="quality-control-report">Quality control report<a class="anchor" aria-label="anchor" href="#quality-control-report"></a>
</h3>
<p>A quality control (QC) report can be performed. It uses several
metrics to help identify outlier samples and/or batch effects, including
end read counts and the C-score itself at all the 7217 genomic
positions. The QC verifies that the coverage is uniform and reproducible
between samples, eliminating the possibility of bias due to sequencing
and outliers</p>
<p>QC can be either performed using a panel of ready to use functions,
which correspond to data visualization, or automatically.</p>
<p>The automatic QC report can be generated using the report_qc
function:</p>
<p>With the ribo_toy example, the column that contains the information
about samples batches is called “run”. The name is thus given to the
<code>library_col</code> parameter.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/report_qc.html">report_qc</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo_toy</span>, library_col <span class="op">=</span> <span class="st">"run"</span><span class="op">)</span></span></code></pre></div>
<p>The QC report includes the following visualizations:</p>
<ul>
<li>End read counts distribution by sample (boxplot_count)</li>
<li>Relative log coverage (i.e., end read count) by sample
(plot_rle)</li>
<li>RNA fraction of end read counts per sample
(plot_count_fraction)</li>
<li>Heatmap summarizing the correlation matrix of the end read counts
(plot_heatmap_corr)</li>
<li>Correspondence analysis of the end read counts (plot_coa)</li>
<li>Principal component analysis (plot_pca)</li>
</ul>
</div>
<div class="section level3">
<h3 id="batch-effect-identification">Batch effect identification<a class="anchor" aria-label="anchor" href="#batch-effect-identification"></a>
</h3>
<p>Technical bias (i.e., batch effect) can be identified by plotting
C-scores at all the genomic positions of the RNA for each sample on a
PCA (see also [Visualization with PCA] for more uses).</p>
<p>Here is an example:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span>ribo <span class="op">=</span> <span class="va">ribo</span>,</span>
<span>         color_col <span class="op">=</span> <span class="st">"run"</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>In this example, the technical replicates RNA1 and RNA2 included in
library 1 and 2 respectively, are distant from each other on the PC1
axis. Moreover, the samples should not be grouped by library or batch.
The following section will resolve this batch effect.</p>
</div>
<div class="section level3">
<h3 id="batch-effect-adjustment">Batch effect adjustment<a class="anchor" aria-label="anchor" href="#batch-effect-adjustment"></a>
</h3>
<p>Batch effect of RiboMethSeq data can be adjusted using the ComBat-seq
method (Paraqindes et al, in revision; Zhang, et al, 2020). The
rRMSAnalyzer package includes a wrapper (adjust_bias) to perform
ComBat-seq adjustment which provides a new RiboClass with adjusted read
end count values and C-scores automatically recalculated with the same
setup parameters.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/adjust_bias.html">adjust_bias</a></span><span class="op">(</span><span class="va">ribo</span>, batch <span class="op">=</span> <span class="st">"run"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Found 2 batches</span></span>
<span><span class="co">#&gt; Using null model in ComBat-seq.</span></span>
<span><span class="co">#&gt; Adjusting for 0 covariate(s) or covariate level(s)</span></span>
<span><span class="co">#&gt; Estimating dispersions</span></span>
<span><span class="co">#&gt; Fitting the GLM model</span></span>
<span><span class="co">#&gt; Shrinkage off - using GLM estimates for parameters</span></span>
<span><span class="co">#&gt; Adjusting the data</span></span>
<span><span class="co">#&gt; Recomputing c-score with the following parameters :</span></span>
<span><span class="co">#&gt; - C-score method : median</span></span>
<span><span class="co">#&gt; - Flanking window : 6</span></span></code></pre></div>
<p>Batch effect adjustment can be verified using the plot_pca function
on the new RiboClass:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="st">"run"</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>After batch effect adjustment using ComBat-seq method, the two
technical replicates RNA1 and RNA2 show reduced dispersion, and the
samples are separated on the PCA axes independently of the library they
belong to.</p>
</div>
</div>
<div class="section level2">
<h2 id="verifying-local-coverage">Verifying local coverage<a class="anchor" aria-label="anchor" href="#verifying-local-coverage"></a>
</h2>
<p>The C-score depends on the local coverage around the site. For a
simple visualization of the read end count coverage of the flanking
region of a given genomic position of interest, use the plot_counts_env
function. Two plotting methods are available:</p>
<ol style="list-style-type: decimal">
<li>
<strong>display all samples</strong> (default parameters): by
displaying the end read count coverage of all samples using a boxplot at
the +6/-6 genomic position relative to the genomic position of interest
(green box). Such a plot is automatically used when number of samples in
up to x. The median of the read end count is shown as a dashed red
line.</li>
</ol>
<p>Here is an example:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ribo_toy"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_counts_env.html">plot_counts_env</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="st">"NR_046235.3_5.8S"</span>,<span class="fl">14</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<ol start="2" style="list-style-type: decimal">
<li>
<strong>sample of interest</strong>: by displaying the end read
count coverage of the sample of interest only using the profile at the
+6/-6 genomic position relative to the genomic position of interest
(vertical green line). The median of the read end count is displayed as
a horizontal dotted red line.</li>
</ol>
<p>Here is an example:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_counts_env.html">plot_counts_env</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="st">"NR_046235.3_5.8S"</span>,<span class="fl">14</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S1"</span>,<span class="st">"S2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="sample-manipulation">Sample manipulation<a class="anchor" aria-label="anchor" href="#sample-manipulation"></a>
</h2>
<div class="section level3">
<h3 id="keep-or-remove-samples">Keep or remove samples<a class="anchor" aria-label="anchor" href="#keep-or-remove-samples"></a>
</h3>
<p>A sample subset can be easily analyzed by specifying which samples to
keep or which to remove. The user can then create a new RiboClass object
containing the data and metadata of the samples of interest. In both
cases, only the metadata of the remaining samples are kept in the
RiboClass object, so no manual updating is required.</p>
<p>Here is an example of how to create a new RiboClass by retaining two
samples of interest (“S1” and “S2”):</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_2samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/keep_ribo_samples.html">keep_ribo_samples</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S1"</span>,<span class="st">"S2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ribo_2samples</span><span class="op">)</span></span>
<span><span class="co">#&gt; a RiboClass with 2 samples and 4 RNA(s) :</span></span>
<span><span class="co">#&gt; Name : NR_023363.1_5S, length : 121</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_5.8S, length : 157</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_18S, length : 1869</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_28S, length : 5070</span></span></code></pre></div>
<p>Here is an example to generate a new RiboClass by removing two
samples (“S1” and “S2”):</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_removed_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_ribo_samples.html">remove_ribo_samples</a></span><span class="op">(</span><span class="va">ribo</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S1"</span>,<span class="st">"S1"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ribo_removed_samples</span><span class="op">)</span></span>
<span><span class="co">#&gt; a RiboClass with 13 samples and 4 RNA(s) :</span></span>
<span><span class="co">#&gt; Name : NR_023363.1_5S, length : 121</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_5.8S, length : 157</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_18S, length : 1869</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_28S, length : 5070</span></span></code></pre></div>
<p>In both cases, only the remaining samples’ metadata are kept in the
RiboClass object. There is no need to update it manually.</p>
</div>
</div>
<div class="section level2">
<h2 id="rna-manipulation-and-annotation">RNA manipulation and annotation<a class="anchor" aria-label="anchor" href="#rna-manipulation-and-annotation"></a>
</h2>
<div class="section level3">
<h3 id="rna-manipulation">RNA manipulation<a class="anchor" aria-label="anchor" href="#rna-manipulation"></a>
</h3>
<div class="section level4">
<h4 id="remove-rna">Remove RNA<a class="anchor" aria-label="anchor" href="#remove-rna"></a>
</h4>
<p>A subset of RNA can be easily analyzed by specifying the RNA to be
removed. The user can thus create a new RiboClass object containing the
data of the RNAs of interest, without affecting the metadata of the
samples.</p>
<p>Here is an example where the RNA 5S is removed:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_rna.html">remove_rna</a></span><span class="op">(</span><span class="va">ribo</span>, rna_to_remove <span class="op">=</span> <span class="st">"NR_023363.1_5S"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ribo_adjusted</span><span class="op">)</span></span>
<span><span class="co">#&gt; a RiboClass with 14 samples and 3 RNA(s) :</span></span>
<span><span class="co">#&gt; Name : NR_046235.3_5.8S, length : 157</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_18S, length : 1869</span></span>
<span><span class="co">#&gt;  Name : NR_046235.3_28S, length : 5070</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="rename-rna">Rename RNA<a class="anchor" aria-label="anchor" href="#rename-rna"></a>
</h4>
<p>The annotation of rRNA 2’Ome sites using the lists provided by this
package requires the use of specific RNA names.</p>
<p>Here is an example to check if the RNA names provided by the user in
the RiboClass match the ones used by this package :</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"human_methylated"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"human_methylated's rna names: "</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">human_methylated</span><span class="op">$</span><span class="va">rRNA</span><span class="op">)</span>,<span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; human_methylated's rna names:  5.8S 18S 28S</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"ribo's rna names: "</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">ribo_adjusted</span><span class="op">$</span><span class="va">rna_names</span><span class="op">$</span><span class="va">current_name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; ribo's rna names:  NR_046235.3_5.8S NR_046235.3_18S NR_046235.3_28S</span></span></code></pre></div>
<p>In this example, the names are different and need to be updated
before annotation.</p>
<p>The rename_rna function automatically updates the rRNA names given by
the rRNA size order:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rename_rna.html">rename_rna</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                            new_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"5.8S"</span>, <span class="st">"18S"</span>, <span class="st">"28S"</span><span class="op">)</span><span class="op">)</span> </span>
<span>                            <span class="co"># from the shortest RNA in our RiboClass to the longest.</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="annotation-of-rna-2ome-sites">Annotation of RNA 2’Ome sites<a class="anchor" aria-label="anchor" href="#annotation-of-rna-2ome-sites"></a>
</h3>
<p>The rRMSAnalyzer package calculates a C-score for each genomic
position of the RNAs. However, not every sites of the reference 2’Ome
RNA is necessarily methylated. Therefore, it is expected that the user
provides a list of potentially methylated sites of interest, called
“annotated sites”. This will make it easier to restrict further analysis
to this list of sites.</p>
<div class="section level4">
<h4 id="included-annotation-human-2ome-rrna-sites">Included annotation : Human 2’Ome rRNA sites<a class="anchor" aria-label="anchor" href="#included-annotation-human-2ome-rrna-sites"></a>
</h4>
<p>By default, rRMSAnalyzer package includes two dataframes containing
the positions and the annotations of the human rRNA 2’Ome sites:</p>
<ul>
<li><p>human_methylated: a dataframe, containing the 112 known 2’Ome
sites for the human rRNAs.</p></li>
<li><p>human_suspected: a dataframe, containing the 17 sites that are
putative 2’Ome sites for the human rRNAs, as described in the
litterature.</p></li>
</ul>
</div>
<div class="section level4">
<h4 id="customize-2ome-sites-annotations">Customize 2’Ome sites annotations<a class="anchor" aria-label="anchor" href="#customize-2ome-sites-annotations"></a>
</h4>
<p>Instead of using the list of human rRNA 2’Ome sites provided by the
rRMSAnalyzer package, the user can provide their own list of annotated
sites which will be attached to the RiboClass object using the
annotate_site function (see 10.3 section : annotation of RNA sites).</p>
<p>This annotate_site function expects an “annot” parameter which must
be a dataframe object that contains the following three mandatory
columns :</p>
<ul>
<li><p><strong>RNA name</strong>: the name of the RNA, matching the RNA
name of the RiboClass.</p></li>
<li><p><strong>Position on RNA</strong>: the number of the position on
the RNA.</p></li>
<li><p><strong>Nomenclature</strong>: the name given to the site of
interest.</p></li>
</ul>
<p>You can see an example below :</p>
<table class="table">
<thead><tr class="header">
<th>Position</th>
<th>rRNA</th>
<th>Nomenclature</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>15</td>
<td>5.8S</td>
<td>Um14</td>
</tr>
<tr class="even">
<td>76</td>
<td>5.8S</td>
<td>Gm75</td>
</tr>
<tr class="odd">
<td>28</td>
<td>18S</td>
<td>Am27</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="annotate-sites">Annotate RNA sites<a class="anchor" aria-label="anchor" href="#annotate-sites"></a>
</h4>
<p>The 2’Ome sites of interest must be attached to the RiboClass object
for further analysis using the annotate function with either the
provided annotations or custom annotations.</p>
<p>Here is an example using the included human methylated
annotations:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_site.html">annotate_site</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                               annot <span class="op">=</span> <span class="va">human_methylated</span>,</span>
<span>                               anno_rna <span class="op">=</span> <span class="st">"rRNA"</span>,</span>
<span>                               anno_pos <span class="op">=</span> <span class="st">"Position"</span>,</span>
<span>                               anno_value <span class="op">=</span> <span class="st">"Nomenclature"</span><span class="op">)</span></span></code></pre></div>
<p>Note : rRMSAnalyzer will display a warning message when there is a
mismatch between the annot and RiboClass RNA names.</p>
<p>This vignette also provides some explanations on how to create your
own sites annotation dataset with <a href="#customize-2ome-sites-annotations">Customize 2’Ome sites
annotations</a>.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="analytic-functions-for-2ome-profiling">Analytic functions for 2’Ome profiling<a class="anchor" aria-label="anchor" href="#analytic-functions-for-2ome-profiling"></a>
</h2>
<p>To determine whether RNA 2’Ome profiles are different between
conditions and identify the most variable RNA 2’Ome sites, several
functions have been implemented to obtain a ggplot. Moreover, the
parameter <code>only_annotated</code>, which is included in all the plot
related functions, enables the plotting of only the annotated RNA 2’Ome
sites of interest (i.e., with biological relevance) when set to true.
Here is a list of the implemented plots with the function name:</p>
<ul>
<li>principal component analysis using the C-scores of the annotated RNA
2’Ome sites (plot_pca)</li>
<li>heatmap using the C-scores of the annotated RNA 2’Ome sites
(plot_heatmap)</li>
<li>boxplot using the C-scores of the annotated RNA 2’Ome sites in
ascending order of variability (boxplot_count)</li>
</ul>
<p>As an example, the plot_pca function is presented below.</p>
<p>Here is an example comparing samples reflecting different biological
conditions based on the rRNA 2’Ome profile of the provided
human_methylated list:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>         color_col <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>         only_annotated <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p><strong>Note 1</strong> : by default, PC1 and PC2 axes are plotted.
However, the user can choose the PCA axes of interest using the “axes”
parameter.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>         color_col <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>         axes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>, <span class="co">#PC2 and PC3 will be plotted</span></span>
<span>         only_annotated <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p><strong>Note 2</strong> the function returns the complete dudi.pca
object instead of the plot by setting <em>object_only</em> to TRUE:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>         color_col <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>         only_annotated <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>         object_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="analytic-functions-for-site-by-site-comparison-of-2ome-levels">Analytic functions for site-by-site comparison of 2’Ome levels<a class="anchor" aria-label="anchor" href="#analytic-functions-for-site-by-site-comparison-of-2ome-levels"></a>
</h2>
<p>Two cut-offs are currently used to identify RNA sites with
significantly different 2’Ome levelsè between biological conditions
(Marcel et al , NAR Cancer 2021):</p>
<ul>
<li>statistical significance: a p-value &lt; 0.01 using Kruskal-Wallis
test.</li>
<li>biological significance: a ΔC-score (i.e., Median
C-score<sub>max</sub> – Median C-score<sub>min</sub>) &gt; 0.05,
indicating a variation of 5% of the RNA 2’Ome level between the
conditions.</li>
</ul>
<p>The <em>plot_diff_sites</em> function displays only the RNA sites
that meet these two cut-off criteria. It produces a boxplot visualizing
the median C-scores per biological condition and the p-value.</p>
<p>Here is an example:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_diff_sites.html">plot_diff_sites</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,factor_column <span class="op">=</span> <span class="st">"condition"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 0 significant sites found !</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Because no RNA site was found to fulfill the default cut-offs, the p-value cut-off is changed for the following example</span></span>
<span><span class="fu"><a href="../reference/plot_diff_sites.html">plot_diff_sites</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,factor_column <span class="op">=</span> <span class="st">"condition"</span>,p_cutoff <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="co">#&gt; 7 significant sites found !</span></span></code></pre></div>
<p><img src="rRMSAnalyzer_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="exporting-data">Exporting data<a class="anchor" aria-label="anchor" href="#exporting-data"></a>
</h2>
<p>Data can be exported as two different objects.</p>
<div class="section level3">
<h3 id="export-as-a-dataframe">Export as a dataframe<a class="anchor" aria-label="anchor" href="#export-as-a-dataframe"></a>
</h3>
<p>The user can export data as a dataframe using the
<em>extract_data</em> function.</p>
<p>By default, C-scores for all the genomic RNA positions are
exported.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_data.html">extract_data</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                        col <span class="op">=</span> <span class="st">"cscore"</span><span class="op">)</span></span></code></pre></div>
<p>The user can export data related to the subset of annotated RNA 2’Ome
sites by setting the <em>only_annotated</em> parameter to TRUE.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ribo_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_data.html">extract_data</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                        col <span class="op">=</span> <span class="st">"cscore"</span>,</span>
<span>                        only_annotated <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Excerpt from the output dataframe, where S1, S2 and S3 are
samples:</caption>
<thead><tr class="header">
<th align="left">site</th>
<th align="right">S1</th>
<th align="right">S2</th>
<th align="right">S3</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">18S_Am27</td>
<td align="right">0.9783333</td>
<td align="right">0.9745514</td>
<td align="right">0.9810412</td>
</tr>
<tr class="even">
<td align="left">18S_Am99</td>
<td align="right">0.9680204</td>
<td align="right">0.9646470</td>
<td align="right">0.9724150</td>
</tr>
<tr class="odd">
<td align="left">18S_Um116</td>
<td align="right">0.9276274</td>
<td align="right">0.9470968</td>
<td align="right">0.9407083</td>
</tr>
<tr class="even">
<td align="left">18S_Um121</td>
<td align="right">0.9630216</td>
<td align="right">0.9684459</td>
<td align="right">0.9691576</td>
</tr>
<tr class="odd">
<td align="left">18S_Am159</td>
<td align="right">0.9629730</td>
<td align="right">0.9620986</td>
<td align="right">0.9686766</td>
</tr>
<tr class="even">
<td align="left">18S_Am166</td>
<td align="right">0.9809492</td>
<td align="right">0.9744627</td>
<td align="right">0.9744998</td>
</tr>
<tr class="odd">
<td align="left">18S_Um172</td>
<td align="right">0.9510189</td>
<td align="right">0.9400922</td>
<td align="right">0.9517981</td>
</tr>
<tr class="even">
<td align="left">18S_Cm174</td>
<td align="right">0.9119119</td>
<td align="right">0.8668456</td>
<td align="right">0.8806886</td>
</tr>
<tr class="odd">
<td align="left">18S_Um354</td>
<td align="right">0.9709273</td>
<td align="right">0.9722054</td>
<td align="right">0.9756174</td>
</tr>
<tr class="even">
<td align="left">18S_Um428</td>
<td align="right">0.9268668</td>
<td align="right">0.9276018</td>
<td align="right">0.9504224</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="export-as-a-ggplot-ready-dataframe">Export as a ggplot-ready dataframe<a class="anchor" aria-label="anchor" href="#export-as-a-ggplot-ready-dataframe"></a>
</h3>
<p>The user can export data as a ggplot-ready dataframe using the
format_to_plot function.</p>
<p>By default, C-scores for all the genomic RNA positions are exported.
The user can export additional information contained in the metadata by
specifying the name of the column of interest. The user can export
information related to the subset of annotated RNA 2’Ome sites by
setting the <em>only_annotated</em> parameter to True.</p>
<p>Here is an example of ggplot-ready dataframe containing the C-scores
of all the genomic RNA positions as well as the condition related to the
particular sample of interest:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ggplot_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_to_plot.html">format_to_plot</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,<span class="st">"condition"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Excerpt from the output ggplot-ready dataframe</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="left">site</th>
<th align="right">cscore</th>
<th align="left">condition</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">501</td>
<td align="left">RNA1</td>
<td align="left">18S_0501</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">502</td>
<td align="left">RNA1</td>
<td align="left">18S_0502</td>
<td align="right">0.2601804</td>
<td align="left">RNA ref</td>
</tr>
<tr class="odd">
<td align="left">503</td>
<td align="left">RNA1</td>
<td align="left">18S_0503</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">504</td>
<td align="left">RNA1</td>
<td align="left">18S_0504</td>
<td align="right">0.3179712</td>
<td align="left">RNA ref</td>
</tr>
<tr class="odd">
<td align="left">505</td>
<td align="left">RNA1</td>
<td align="left">18S_0505</td>
<td align="right">0.5132894</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">506</td>
<td align="left">RNA1</td>
<td align="left">18S_0506</td>
<td align="right">0.6354548</td>
<td align="left">RNA ref</td>
</tr>
<tr class="odd">
<td align="left">507</td>
<td align="left">RNA1</td>
<td align="left">18S_0507</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">508</td>
<td align="left">RNA1</td>
<td align="left">18S_0508</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="odd">
<td align="left">509</td>
<td align="left">RNA1</td>
<td align="left">18S_0509</td>
<td align="right">0.0000000</td>
<td align="left">RNA ref</td>
</tr>
<tr class="even">
<td align="left">510</td>
<td align="left">RNA1</td>
<td align="left">18S_0510</td>
<td align="right">0.9380812</td>
<td align="left">RNA ref</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="export-as-a-dataframe-by-condition">Export as a dataframe by condition<a class="anchor" aria-label="anchor" href="#export-as-a-dataframe-by-condition"></a>
</h3>
<p>The <em>mean_samples_by_condition</em> function provdes the
summarized values per modality of a variable in the metadata
dataframe.</p>
<p>By default, the function provides for all the genomic RNA positions
the name of the position, the mean and standard deviation (sd) of the
C-scores. By using the value=”count” parameter, it is als possible to
calculate these values from the read end counts. Calculations can be
limited to only annotated RNA 2’Ome sites by setting the only_annotated
parameter to TRUE.</p>
<p>Here is an example of dataframe showing the summarized C-scores per
modality of the “condition” variable for all the genomic RNA
positions:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mean_tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mean_samples_by_conditon.html">mean_samples_by_conditon</a></span><span class="op">(</span><span class="va">ribo_adjusted</span>,</span>
<span>                                    value <span class="op">=</span> <span class="st">"cscore"</span>,</span>
<span>                                    metadata_condition <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                                    only_annotated <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Excerpt from the output dataframe by condition</caption>
<thead><tr class="header">
<th align="left">site</th>
<th align="left">condition</th>
<th align="right">mean</th>
<th align="right">sd</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">18S_Am1031</td>
<td align="left">RNA ref</td>
<td align="right">0.9749922</td>
<td align="right">0.0047461</td>
</tr>
<tr class="even">
<td align="left">18S_Am1031</td>
<td align="left">cond1</td>
<td align="right">0.9703699</td>
<td align="right">0.0032725</td>
</tr>
<tr class="odd">
<td align="left">18S_Am1031</td>
<td align="left">cond2</td>
<td align="right">0.9723714</td>
<td align="right">0.0057723</td>
</tr>
<tr class="even">
<td align="left">18S_Am1383</td>
<td align="left">RNA ref</td>
<td align="right">0.9796876</td>
<td align="right">0.0035939</td>
</tr>
<tr class="odd">
<td align="left">18S_Am1383</td>
<td align="left">cond1</td>
<td align="right">0.9765212</td>
<td align="right">0.0027083</td>
</tr>
<tr class="even">
<td align="left">18S_Am1383</td>
<td align="left">cond2</td>
<td align="right">0.9779684</td>
<td align="right">0.0028759</td>
</tr>
<tr class="odd">
<td align="left">18S_Am159</td>
<td align="left">RNA ref</td>
<td align="right">0.9684126</td>
<td align="right">0.0022025</td>
</tr>
<tr class="even">
<td align="left">18S_Am159</td>
<td align="left">cond1</td>
<td align="right">0.9609920</td>
<td align="right">0.0054882</td>
</tr>
<tr class="odd">
<td align="left">18S_Am159</td>
<td align="left">cond2</td>
<td align="right">0.9623362</td>
<td align="right">0.0065239</td>
</tr>
<tr class="even">
<td align="left">18S_Am166</td>
<td align="left">RNA ref</td>
<td align="right">0.9826490</td>
<td align="right">0.0033169</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">Sessioninfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] rRMSAnalyzer_2.0.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] ade4_1.7-22             tidyselect_1.2.0        dplyr_1.1.2            </span></span>
<span><span class="co">#&gt;   [4] farver_2.1.1            blob_1.2.4              bitops_1.0-7           </span></span>
<span><span class="co">#&gt;   [7] Biostrings_2.68.1       RCurl_1.98-1.12         fastmap_1.1.1          </span></span>
<span><span class="co">#&gt;  [10] XML_3.99-0.14           digest_0.6.31           factoextra_1.0.7       </span></span>
<span><span class="co">#&gt;  [13] lifecycle_1.0.3         survival_3.5-5          KEGGREST_1.40.0        </span></span>
<span><span class="co">#&gt;  [16] RSQLite_2.3.1           magrittr_2.0.3          genefilter_1.82.1      </span></span>
<span><span class="co">#&gt;  [19] compiler_4.3.1          rlang_1.1.1             sass_0.4.6             </span></span>
<span><span class="co">#&gt;  [22] tools_4.3.1             utf8_1.2.3              yaml_2.3.7             </span></span>
<span><span class="co">#&gt;  [25] knitr_1.43              ggsignif_0.6.4          labeling_0.4.2         </span></span>
<span><span class="co">#&gt;  [28] bit_4.0.5               abind_1.4-5             BiocParallel_1.34.2    </span></span>
<span><span class="co">#&gt;  [31] withr_2.5.0             purrr_1.0.1             BiocGenerics_0.46.0    </span></span>
<span><span class="co">#&gt;  [34] desc_1.4.2              grid_4.3.1              stats4_4.3.1           </span></span>
<span><span class="co">#&gt;  [37] fansi_1.0.4             ggpubr_0.6.0            xtable_1.8-4           </span></span>
<span><span class="co">#&gt;  [40] colorspace_2.1-0        edgeR_3.42.4            ggplot2_3.4.2          </span></span>
<span><span class="co">#&gt;  [43] scales_1.2.1            MASS_7.3-60             cli_3.6.1              </span></span>
<span><span class="co">#&gt;  [46] crayon_1.5.2            rmarkdown_2.22          ragg_1.2.5             </span></span>
<span><span class="co">#&gt;  [49] generics_0.1.3          httr_1.4.6              DBI_1.1.3              </span></span>
<span><span class="co">#&gt;  [52] cachem_1.0.8            stringr_1.5.0           splines_4.3.1          </span></span>
<span><span class="co">#&gt;  [55] zlibbioc_1.46.0         parallel_4.3.1          AnnotationDbi_1.62.1   </span></span>
<span><span class="co">#&gt;  [58] XVector_0.40.0          matrixStats_1.0.0       vctrs_0.6.3            </span></span>
<span><span class="co">#&gt;  [61] Matrix_1.5-4.1          jsonlite_1.8.5          sva_3.48.0             </span></span>
<span><span class="co">#&gt;  [64] carData_3.0-5           car_3.1-2               IRanges_2.34.0         </span></span>
<span><span class="co">#&gt;  [67] S4Vectors_0.38.1        bit64_4.0.5             rstatix_0.7.2          </span></span>
<span><span class="co">#&gt;  [70] ggrepel_0.9.3           systemfonts_1.0.4       locfit_1.5-9.8         </span></span>
<span><span class="co">#&gt;  [73] limma_3.56.2            tidyr_1.3.0             jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt;  [76] annotate_1.78.0         glue_1.6.2              pkgdown_2.0.7          </span></span>
<span><span class="co">#&gt;  [79] codetools_0.2-19        stringi_1.7.12          gtable_0.3.3           </span></span>
<span><span class="co">#&gt;  [82] GenomeInfoDb_1.36.1     munsell_0.5.0           tibble_3.2.1           </span></span>
<span><span class="co">#&gt;  [85] pillar_1.9.0            htmltools_0.5.5         GenomeInfoDbData_1.2.10</span></span>
<span><span class="co">#&gt;  [88] R6_2.5.1                textshaping_0.3.6       rprojroot_2.0.3        </span></span>
<span><span class="co">#&gt;  [91] lattice_0.21-8          evaluate_0.21           Biobase_2.60.0         </span></span>
<span><span class="co">#&gt;  [94] highr_0.10              png_0.1-8               backports_1.4.1        </span></span>
<span><span class="co">#&gt;  [97] memoise_2.0.1           broom_1.0.5             bslib_0.5.0            </span></span>
<span><span class="co">#&gt; [100] Rcpp_1.0.10             nlme_3.1-162            mgcv_1.8-42            </span></span>
<span><span class="co">#&gt; [103] xfun_0.39               fs_1.6.2                MatrixGenerics_1.12.2  </span></span>
<span><span class="co">#&gt; [106] pkgconfig_2.0.3</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-birkedal2014" class="csl-entry">
Birkedal, Ulf, Mikkel Christensen-Dalsgaard, Nicolai Krogh,
Radhakrishnan Sabarinathan, Jan Gorodkin, and Henrik Nielsen. 2014.
<span>“Profiling of Ribose Methylations in RNA by High-Throughput
Sequencing.”</span> <em>Angewandte Chemie International Edition</em>,
November, n/a–. <a href="https://doi.org/10.1002/anie.201408362" class="external-link">https://doi.org/10.1002/anie.201408362</a>.
</div>
<div id="ref-Erales2017" class="csl-entry">
Erales, Jenny, Virginie Marchand, Baptiste Panthu, Sandra Gillot,
Stéphane Belin, Sandra E. Ghayad, Maxime Garcia, et al. 2017.
<span>“Evidence for rRNA 2&amp;#x2032;-o-Methylation Plasticity: Control
of Intrinsic Translational Capabilities of Human Ribosomes.”</span>
<em>Proceedings of the National Academy of Sciences</em> 114 (49):
12934–39. <a href="https://doi.org/10.1073/pnas.1707674114" class="external-link">https://doi.org/10.1073/pnas.1707674114</a>.
</div>
<div id="ref-Marcel2021" class="csl-entry">
Marcel, Virginie, Janice Kielbassa, Virginie Marchand, Kundhavai S
Natchiar, Hermes Paraqindes, Flora Nguyen Van Long, Lilia Ayadi, et al.
2021. <span>“<span class="nocase">Ribosomal RNA 2’ O-methylation as a
novel layer of inter-tumour heterogeneity in breast
cancer</span>.”</span> <em>NAR Cancer</em> 3 (1). <a href="https://doi.org/10.1093/narcan/zcab006" class="external-link">https://doi.org/10.1093/narcan/zcab006</a>.
</div>
<div id="ref-Marchand2016" class="csl-entry">
Marchand, Virginie, Florence Blanloeil-Oillo, Mark Helm, and Yuri
Motorin. 2016. <span>“<span class="nocase">Illumina-based RiboMethSeq
approach for mapping of 2’-O-Me residues in RNA</span>.”</span>
<em>Nucleic Acids Research</em> 44 (16): e135–35. <a href="https://doi.org/10.1093/nar/gkw547" class="external-link">https://doi.org/10.1093/nar/gkw547</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Théo Combe, Hermes Paraqindes, Janice Kielbassa, Virginie Marcel, Émilie Thomas, Anthony Ferrari, Fondation Synergie Lyon Cancer, INCA, SIRIC program, LabEX program, Ligue Nationale Contre le Cancer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
