---
title: "RiboSTORY: RiboMeth-Seq Tools – Pre-Analytic report"
date: "`r Sys.time()`"
output:
  rmdformats::robobook:
    self_contained: true
    number_sections: true
params:
  project_name : "mon beau projet"
  data_path: "/home/theo/Counts/"
  
  metadata_path: "/home/theo/metadata.csv"
  metadata_sep: ","
  filename_col: "filename"
  sample_name_col : "Sample.name.on.tubes"
  library_col: "Librairies.à.réaliser.ensemble"
  biological_condition_col : "SAMPLE..ORIGIN"
    
---
<style type="text/css">
  body{
 font-family: Helvetica;    
}

details{
background-color: #E4F3F3;
}
</style>


```{r setup, include=FALSE}
library(knitr)
library(rmdformats)
library(RiboMethSeq)
library(DT)
library(ade4)
## Global options
options(max.print="75")
opts_chunk$set(echo=FALSE,
	             cache=TRUE,
               prompt=FALSE,
               tidy=TRUE,
               comment=NA,
               message=FALSE,
               warning=FALSE)
opts_knit$set(width=75)

group_to_show <- params$biological_condition_col

metadata <- read.csv(params$metadata_path, sep = params$metadata_sep)


ribo <- RiboMethSeq::read_counts(params$data_path,
                                 metadata = metadata,
                                 metadata_sep = ",",
                                 metadata_filename_col = params$filename_col,
                                 metadata_sample_name_col = params$sample_name_col)



ribo$rna_names$new_rna_name = c("5.8S","18S","28S","5S")
ribo <- update_riboclass_rna_names(ribo)

ribo <- RiboMethSeq::calculate_score(ribo, use_multithreads = T)


```

# Objectives {-}
2’Ome is a chemical modification of the ribosomal RNA that is altered in several biological processes affecting translational control [(Jaafar et al, 2021 – PMID 34440717)](https://www.mdpi.com/2073-4409/10/8/1948). rRNA 2’Ome can be analysed using RiboMeth-Seq approach [(Birkedal et al, 2015 - PMID 25417815)](https://pubmed.ncbi.nlm.nih.gov/25417815/). <br>
The RiboSTORY package is dedicated to pre-analytic analysis of RiboMeth-Seq data. A schematic representation of the RiboSTORY package is given at the end of this report (link). It allows to:
<br>
(1) calculate the rRNA 2’Ome level (i.e., C-score) from RiboMETH-seq data (i.e., end read count) <br>
(2) report quality control metrics to identify putative outlier sample <br>
(3) give a pre-analytic overview of the biological data 

For a better clarity, some explanations are folded by default. You can by clicking on any title containing the ℹ️ symbol.

<details>
<summary><b> ℹ️ What is a C-score ? (click here !)</b></summary>

The C-score corresponds to the 2’Ome level at a rRNA position known to be methylated. The C-score represents a drop in end read coverage at a given position compared to the environmental coverage as described by Birkedal et al, 2015 [(PMID 25417815)](https://pubmed.ncbi.nlm.nih.gov/25417815/). 

</details>


# Project summary

\---------------------------- <br>
**Project name:** `r params$project_name` <br>
**User name:** `r Sys.getenv("USERNAME")` <br>
**Date/Time:** `r Sys.time()` <br>
\---------------------------- <br>

## General informations

Number of uploaded samples: **`r length(ribo[[1]])`** samples. <br>
The following (r)RNA were used by your coverage tool: **`r as.character(ribo[["rna_names"]][["original_name"]])`** <br>
There are **`r length(ribo[[1]][[1]][[1]])`** genomic positions.

## Metadata
```{r}
DT::datatable(ribo[[2]])
```

# Quality Control 
The quality control section reports different metrics to help in identifying outlier sample and/or batch effect. The quality control is performed using non-biological noise that corresponds to both end read counts (used to calculate the C-score) and the C-score itself at all the `r length(ribo[[1]][[1]][[1]])` genomic positions. 

## Quality control based on the end read counts at all genomic positions 
### Coverage

The coverage corresponds to the number of end read counts at each individual genomic position. 

First, by comparing the C-score at a given genomic position depending on the number of end read counts in its local environment, it appears that a minimum coverage of 100 end read counts is required to reach a robust C-score. **Here is a boxplot given the distribution of the absolute number of end read counts at each genomic positions for all individual samples.** The blue line gives the minimal 100 end read counts. Median below the blue line identifies putative outlier sample.

```{r boxplot}
plot(RiboMethSeq::plot_boxplot_count(ribo))
```
Second, sample that presents a lower or higher coverage might be considered as outlier since C-score robustness is dependent on the coverage. **Here is a Relative Log Coverage plot given the variation of the end read counts to the serie/sample (?) median for each individual sample.** The blue lines give median +/- 2mad variation. Median outside the blue lines identifies putative outlier sample.
```{r}
RiboMethSeq::plot_RLE(ribo,"Count",show_outlier = T)
```

### Coverage profile

The coverage profile corresponds to the number of end read counts at each individual genomic position along the genomic sequence. 
 
rRNA sequences are enriched in GC-stretches and in nucleotide repeats thus inducing a bias during read alignment. The coverage profile is thus typical of the rRNA sequence. **Here is a coverage profile plot given for the 4 rRNAs, the absolute number of end read counts at each genomic position being plotted along the rRNA sequences for each sample.** Distinct coverage profile identifies putative outlier sample. To help in identifying distinct coverage profile, two methods are given after these plots.  

```{r}
plot_count_profile_test(ribo)
```


### Correlation-based distance heatmap {.tabset}

First, to help in identifying distinct coverage profile, coverage profiles are compared between two samples in the whole series. Two-by-two sample correlation of the coverage profile is calculated using x. **Here is given a distance heatmap that summarizes of the correlation scores in the entire series either as clusterized or non-clusterized (tab)**. A perfect correlation is given by 0 (white) and no correlation by 1 (red). 

#### Clusterized {-}
```{r heatmap_corr}
RiboMethSeq::plot_heatmap_corr(ribo,"Count",NA,"samplename")
```

#### non-clusterized {-}

```{r heatmap_corr_tri}
RiboMethSeq::plot_heatmap_corr(ribo,"Count",NA,"samplename",use_triangle=T)
```

Second, to identify lack of two-by-two correlation between coverage profile and identify putative outlier sample, distribution of the correlation data is performed. **Here is a summarized boxplot of the two-by-two correlation scores of the entire series**. Putative outlier samples are represented as open circles (Q1 – 1.5*IQR).

### Coverage profile: serie comparison

First, to help in identifying distinct coverage profile, coverage profiles are compared between the whole series. **Here is given a Component Analysis (COA) plot that illustrates the distance between each sample based on its coverage profile**. Distant sample is a putative outlier. 

```{r}
plot_fca(ribo,col_for_color = params$library_col)
```

Second, to identify lack of correlation between coverage profile inside the entire series and identify putative outlier sample, distance from a given sample to the origin is calculated. Here is a plot given the distance to the origin for each sample. The blue line give median + 3mad variation. Distance to origin above the blue line identifies putative outlier sample.

## Quality control based on the C-scores at all genomic positions

The C-score corresponds to the normalization of the end read count regarding the local environment at each genomic position. Since only the C-score of the known 2’Ome position are further extracted to give the rRNA 2’Ome level and biological analysis (< 1.5%), the C-score at all genomic position is then used in this second QC part. 

### C-score: individual sample

Since more than 1.5% of the C-score corresponds to non-biological noise, normalisation during C-score calculation should limit the dispersion of the C-score at all genomic positions in a single sample. **Here is given a Relative Log C-score plot for each sample**. Blue lines indicate median +/- 2mad. Median of boxplot outside these lines identifies putative outlier sample.

```{r}
RiboMethSeq::plot_RLE(ribo,"cscore_median",show_outlier = T)

```

### C-score: serie comparison

Since more than 1.5% of the C-score corresponds to non-biological noise, normalisation during C-score calculation should limit the dispersion of the samples based on their C-score at all genomic positions in the whole series. **Here is given a Principal Component Analysis (PCA) plot that illustrates the distance between each sample based on its C-score at all genomic positions.** Color indicates distinct library. Moreover, this plot helps in identifying putative batch effect that can be corrected using the inter-normalization option (Combat-seq tool). 

```{r}
RiboMethSeq::plot_PCA(ribo,params$library_col)
```



# Analysis of rRNA 2’Ome



```{r}
load("data/human.methylated.rda")
load("data/human.suspected.rda")
human.methylated$named_position <- paste0(human.methylated$rRNA,"_",formatC(human.methylated$Position,width = 4,flag = "0"))
human.methylated$renamed_position <- paste0(human.methylated$rRNA,"_",human.methylated$Nomenclature)

human.suspected$named_position <- paste0(human.suspected$rRNA,"_",formatC(human.suspected$Position,width = 4,flag = "0"))
human.suspected$renamed_position <- paste0(human.suspected$rRNA,"_",human.suspected$Nomenclature)

human_all <- bind_rows(human.methylated,human.suspected)

ribo_meth <- RiboMethSeq::subset_ribo(ribo,human.methylated)

ribo_sus <- RiboMethSeq::subset_ribo(ribo,human.suspected)

ribo_meth_sus <- RiboMethSeq::subset_ribo(ribo,human_all)
```
The C-score corresponding to the known rRNA 2’Ome sites are extracting from all the genomic positions to perform the biological analysis. The number of well identified and mapped rRNA 2’Ome sites is in constant evolution since some studies identifies novel rRNA 2’Ome sites depending on particular biological context (Jaafar et al, 2021 – PMID 34440717). 
Different lists of rRNA 2’Ome sites can thus be used to extract C-score of interest. The following lists of rRNA 2’Ome sites are used in this analysis: <br>

* number of known 2’Ome sites (i.e., well established): **`r length(human.methylated[,"Position"])`**
* number of suspected 2’Ome sites (i.e., putative sites): **`r length(human.suspected[,"Position"])`**

Here is a table, which summarizes the known rRNA 2’Ome sites.
```{r}
DT::datatable(human.methylated)
```



## Analysis of rRNA 2’Ome profile {.tabset}

In this first analysis section, the alteration in rRNA 2’Ome is analysed globally, by comparing the rRNA 2’Ome profile encompassing the 2’Ome level of all the methylation site of interest.

Here is given a Principal Component Analysis (PCA) plot that illustrates the distance between each sample based on its rRNA 2’Ome profile. The percentage in dimension 1 and 2 give the distance between samples, high percentage reflecting high distance. Three PCA are given based on rRNA 2’Ome profiles of the known sites, the suspected sites and the known+suspected sites (tab). Color indicates the different biological groups. Moreover, this plot helps in identifying putative batch effect that can be corrected using the inter-normalization option (Combat-seq tool). 

### Known methylation sites {-}

```{r}
RiboMethSeq::plot_PCA(ribo_meth,group_to_show)
```

### suspected methylation sites {-}
```{r}
RiboMethSeq::plot_PCA(ribo_sus,group_to_show)
```

### suspected + known methylation sites {-}
```{r}
RiboMethSeq::plot_PCA(ribo_meth_sus,group_to_show)
```

## Analysis of rRNA 2’Ome profile (heatmap) {.tabset}

Here is given an heatmap that illustrates the distance between each sample based on its rRNA 2’Ome profile. Three heatmap are given based on rRNA 2’Ome profiles of the known sites, the suspected sites and the known+suspected sites (tab). Color indicates the different biological groups. C-score is represented by colored square: 0 = no rRNA 2’Ome in a particular sample; 1 = all rRNA 2’Ome in a particular sample; ]0:1[ = a mix of un-methylated and methylated rRNA in a particular sample. Moreover, this plot helps in identifying putative batch effect that can be corrected using the inter-normalization option (Combat-seq tool). 


### Known methylation sites {-}
```{r fig.height=15,fig.width=10}
RiboMethSeq::plot_heatmap(ribo_meth,cols_for_annotation = c(group_to_show,params$library_col))
```

### Suspected methylation sites {-}
```{r}
RiboMethSeq::plot_heatmap(ribo_sus,c(group_to_show,params$library_col))
```

### Known + suspected methylation sites {-}
```{r fig.height=18,fig.width=13}
RiboMethSeq::plot_heatmap(ribo_meth_sus,
                          c(group_to_show,params$library_col))
```




## Analysis of rRNA 2’Ome variability

```{r}
ribo_matrix <- aggregate_samples_by_col(ribo_meth[["counts"]],"cscore_median")
ribo_variable_positions <- get_most_or_less_variant(ribo_matrix)
ribo_variable <- subset_ribo(ribo_meth, ribo_variable_positions)
```


In this second analysis section, the objective is (1) to identify the most variable rRNA 2’Ome sites in a particular biological condition as described in Marcel et al, 2020 (PMID 34316693), and (2) to determine whether the minimal rRNA 2’Ome profile composed of theses variable sites associate with biological context differentially from the whole rRNA 2’Ome profile. This analysis is restricted to the known rRNA 2’Ome sites.

First, to identify the most variable rRNA 2’Ome sites, variability of the C-score among the biological samples is analysed. **Here is given a boxplot representing the C-score values of all samples for each individual site that are representing in ascending IQR (interquartile range) order (C-score variability tab), where the blue line represents the proposed limit between “stable” (left) and “variable” (right) rRNA 2’Ome sites.** <br>

```{r,fig.width=13}
# reorder on IQR
plot_sites_by_IQR(ribo_meth,plot = "boxplot",variance = "iqr")
```

In addition, a second plot is given representing the IQR number for each individual site ordered in ascending IQR number (IQR tab), where the red line encompass the maximum number of the first IQR and allowing to identify variable sites.  

```{r}
plot_sites_by_IQR(ribo_meth,variance = "iqr")

```


Second, to determine whether usage of the most variable rRNA 2’Ome sites associate with biological context, previous analyses are performed (cf 3.2) using this minimal set of variable rRNA 2’Ome sites.

Here is given a Principal Component Analysis (PCA) based on rRNA 2’Ome profile composed of the most variable rRNA 2’Ome sites. Color indicates the different biological groups. 

```{r}
plot_PCA(ribo_variable, params$biological_condition_col)
```


Here is given an heatmap . Color indicates the different biological groups. C-score is represented by colored square: 0 = no rRNA 2’Ome in a particular sample; 1 = all rRNA 2’Ome in a particular sample; ]0:1[ = a mix of un-methylated and methylated rRNA in a particular sample. 

```{r}
plot_heatmap(ribo_variable)
```


## Site-by-site analysis of rRNA 2’Ome

In this last analysis section, the alteration in rRNA 2’Ome is analysed site-by-site individually to identify significant alteration in rRNA 2’Ome level at particular site depending on the biological context.

### Site-by-site analysis of all rRNA 2’Ome sites {.tabset}

Mean comparison of C-score between biological conditions is performed for all rRNA 2’Ome sites.

Here is given a line plot given the mean C-score and the standard deviation for each rRNA 2’Ome sites in the different biological context. Two plots are given the known or suspected sites (tab). Color indicates the different biological groups. 
Line plot=> enlever le titre 

```{r}
plot_lineplot(ribo_meth,"cscore_median",params$biological_condition_col)
```


Here is given an histo plot given the mean C-score and the standard deviation in the different biological context for each individual rRNA 2’Ome sites. Two plots are given the known or suspected sites (tab). Color indicates the different biological groups. 

#### Methylated {-}

```{r,fig.width=15,fig.height=10}
RiboMethSeq::plot_barpos(ribo_meth,"cscore_median","SAMPLE..ORIGIN")
```

#### Suspected {-}

```{r,fig.width=15,fig.height=10}
RiboMethSeq::plot_barpos(ribo_sus,"cscore_median","SAMPLE..ORIGIN")
```

```{r}
# First, let's get our more significant sites
  ribo_matrix <- aggregate_samples_by_col(ribo_meth[["counts"]],"cscore_median",position_to_rownames = T)
  kruskal_df <- wrapper.kruskal.test(ribo_meth,order_by_col = "samplename",factor_column = params$biological_condition_col)
significant_sites <- select_most_significant_sites(df_of_kruskal = kruskal_df)

ribo_most_variable <- subset_ribo(ribo_meth, significant_sites)
```


### Site-by-site analysis of all significant rRNA 2’Ome sites {.tabset}

The following sites are the most variant :  **`r as.character(significant_sites)`**


Mean comparison of C-score between biological conditions for significant rRNA 2’Ome sites is shown.

First, mean comparison between biological conditions is performed for all the rRNA 2’Ome sites using non-parametric method of Kruskall Wallis (biological conditions > 2) or Wilcoxon test (biological conditions  = 2)P-values are adjusted using the false discovery rate (fdr) method. . In case of more than 2 biological conditions, a second round of mean comparison is performed for the rRNA 2’Ome sites displaying a significant Kruskall Wallis P-value this time between each biological conditions (pairwise wilcoxo test).  

Here is given a box plot given the C-score in the different biological context for each rRNA sites whose 2’Ome is significantly different between the biological conditions. Two plots are given the known or suspected sites (tab). Color indicates the different biological groups. 

#### methylated {-}
```{r,fig.width=10}
plot_test_wrapper(ribo_meth,values_column = "cscore_median","SAMPLE..ORIGIN")    
```

#### suspected {-}
```{r,fig.width=10}
plot_test_wrapper(ribo_sus,values_column = "cscore_median","SAMPLE..ORIGIN")    
```




# About RiboStory


